// @aws-sdk/client-ecr/dist-cjs/schemas/schemas_0.js
const _A = "Attribute";
const _AD = "AuthorizationData";
const _ADL = "AuthorizationDataList";
const _AECID = "AwsEcrContainerImageDetails";
const _AL = "AttributeList";
const _BBOPE = "BlockedByOrganizationPolicyException";
const _BCLA = "BatchCheckLayerAvailability";
const _BCLAR = "BatchCheckLayerAvailabilityRequest";
const _BCLARa = "BatchCheckLayerAvailabilityResponse";
const _BDI = "BatchDeleteImage";
const _BDIR = "BatchDeleteImageRequest";
const _BDIRa = "BatchDeleteImageResponse";
const _BGI = "BatchGetImage";
const _BGIR = "BatchGetImageRequest";
const _BGIRa = "BatchGetImageResponse";
const _BGRSC = "BatchGetRepositoryScanningConfiguration";
const _BGRSCR = "BatchGetRepositoryScanningConfigurationRequest";
const _BGRSCRa = "BatchGetRepositoryScanningConfigurationResponse";
const _CLU = "CompleteLayerUpload";
const _CLUR = "CompleteLayerUploadRequest";
const _CLURo = "CompleteLayerUploadResponse";
const _CPTCR = "CreatePullThroughCacheRule";
const _CPTCRR = "CreatePullThroughCacheRuleRequest";
const _CPTCRRr = "CreatePullThroughCacheRuleResponse";
const _CR = "CreateRepository";
const _CRCT = "CreateRepositoryCreationTemplate";
const _CRCTR = "CreateRepositoryCreationTemplateRequest";
const _CRCTRr = "CreateRepositoryCreationTemplateResponse";
const _CRR = "CreateRepositoryRequest";
const _CRRr = "CreateRepositoryResponse";
const _CS = "CvssScore";
const _CSA = "CvssScoreAdjustment";
const _CSAL = "CvssScoreAdjustmentList";
const _CSD = "CvssScoreDetails";
const _CSL = "CvssScoreList";
const _DI = "DescribeImages";
const _DIF = "DescribeImagesFilter";
const _DIR = "DescribeImagesRequest";
const _DIRS = "DescribeImageReplicationStatus";
const _DIRSR = "DescribeImageReplicationStatusRequest";
const _DIRSRe = "DescribeImageReplicationStatusResponse";
const _DIRe = "DescribeImagesResponse";
const _DISF = "DescribeImageScanFindings";
const _DISFR = "DescribeImageScanFindingsRequest";
const _DISFRe = "DescribeImageScanFindingsResponse";
const _DISS = "DescribeImageSigningStatus";
const _DISSR = "DescribeImageSigningStatusRequest";
const _DISSRe = "DescribeImageSigningStatusResponse";
const _DLP = "DeleteLifecyclePolicy";
const _DLPR = "DeleteLifecyclePolicyRequest";
const _DLPRe = "DeleteLifecyclePolicyResponse";
const _DPTCR = "DeletePullThroughCacheRule";
const _DPTCRR = "DeletePullThroughCacheRuleRequest";
const _DPTCRRe = "DeletePullThroughCacheRuleResponse";
const _DPTCRRes = "DescribePullThroughCacheRulesRequest";
const _DPTCRResc = "DescribePullThroughCacheRulesResponse";
const _DPTCRe = "DescribePullThroughCacheRules";
const _DPTUE = "DeregisterPullTimeUpdateExclusion";
const _DPTUER = "DeregisterPullTimeUpdateExclusionRequest";
const _DPTUERe = "DeregisterPullTimeUpdateExclusionResponse";
const _DR = "DeleteRepository";
const _DRCT = "DeleteRepositoryCreationTemplate";
const _DRCTR = "DeleteRepositoryCreationTemplateRequest";
const _DRCTRe = "DeleteRepositoryCreationTemplateResponse";
const _DRCTRes = "DescribeRepositoryCreationTemplatesRequest";
const _DRCTResc = "DescribeRepositoryCreationTemplatesResponse";
const _DRCTe = "DescribeRepositoryCreationTemplates";
const _DRP = "DeleteRegistryPolicy";
const _DRPR = "DeleteRegistryPolicyRequest";
const _DRPRe = "DeleteRegistryPolicyResponse";
const _DRPRel = "DeleteRepositoryPolicyRequest";
const _DRPRele = "DeleteRepositoryPolicyResponse";
const _DRPe = "DeleteRepositoryPolicy";
const _DRR = "DeleteRepositoryRequest";
const _DRRe = "DeleteRepositoryResponse";
const _DRRes = "DescribeRegistryRequest";
const _DRResc = "DescribeRegistryResponse";
const _DRRescr = "DescribeRepositoriesRequest";
const _DRRescri = "DescribeRepositoriesResponse";
const _DRe = "DescribeRegistry";
const _DRes = "DescribeRepositories";
const _DSC = "DeleteSigningConfiguration";
const _DSCR = "DeleteSigningConfigurationRequest";
const _DSCRe = "DeleteSigningConfigurationResponse";
const _EAEE = "ExclusionAlreadyExistsException";
const _EC = "EncryptionConfiguration";
const _ECFRCT = "EncryptionConfigurationForRepositoryCreationTemplate";
const _EISF = "EnhancedImageScanFinding";
const _EISFL = "EnhancedImageScanFindingList";
const _ENFE = "ExclusionNotFoundException";
const _EUE = "EmptyUploadException";
const _GAS = "GetAccountSetting";
const _GASR = "GetAccountSettingRequest";
const _GASRe = "GetAccountSettingResponse";
const _GAT = "GetAuthorizationToken";
const _GATR = "GetAuthorizationTokenRequest";
const _GATRe = "GetAuthorizationTokenResponse";
const _GDUFL = "GetDownloadUrlForLayer";
const _GDUFLR = "GetDownloadUrlForLayerRequest";
const _GDUFLRe = "GetDownloadUrlForLayerResponse";
const _GLP = "GetLifecyclePolicy";
const _GLPP = "GetLifecyclePolicyPreview";
const _GLPPR = "GetLifecyclePolicyPreviewRequest";
const _GLPPRe = "GetLifecyclePolicyPreviewResponse";
const _GLPR = "GetLifecyclePolicyRequest";
const _GLPRe = "GetLifecyclePolicyResponse";
const _GRP = "GetRegistryPolicy";
const _GRPR = "GetRegistryPolicyRequest";
const _GRPRe = "GetRegistryPolicyResponse";
const _GRPRet = "GetRepositoryPolicyRequest";
const _GRPRete = "GetRepositoryPolicyResponse";
const _GRPe = "GetRepositoryPolicy";
const _GRSC = "GetRegistryScanningConfiguration";
const _GRSCR = "GetRegistryScanningConfigurationRequest";
const _GRSCRe = "GetRegistryScanningConfigurationResponse";
const _GSC = "GetSigningConfiguration";
const _GSCR = "GetSigningConfigurationRequest";
const _GSCRe = "GetSigningConfigurationResponse";
const _I = "Image";
const _IAE = "ImageArchivedException";
const _IAEE = "ImageAlreadyExistsException";
const _ID = "ImageDetail";
const _IDDNME = "ImageDigestDoesNotMatchException";
const _IDL = "ImageDetailList";
const _IF = "ImageFailure";
const _IFL = "ImageFailureList";
const _II = "ImageIdentifier";
const _IIL = "ImageIdentifierList";
const _IL = "ImageList";
const _ILE = "InvalidLayerException";
const _ILPE = "InvalidLayerPartException";
const _ILU = "InitiateLayerUpload";
const _ILUR = "InitiateLayerUploadRequest";
const _ILURn = "InitiateLayerUploadResponse";
const _INFE = "ImageNotFoundException";
const _IPE = "InvalidParameterException";
const _IR = "ImageReferrer";
const _IRL = "ImageReferrerList";
const _IRS = "ImageReplicationStatus";
const _IRSL = "ImageReplicationStatusList";
const _ISC = "ImageScanningConfiguration";
const _ISCUNSE = "ImageStorageClassUpdateNotSupportedException";
const _ISF = "ImageScanFinding";
const _ISFL = "ImageScanFindingList";
const _ISFS = "ImageScanFindingsSummary";
const _ISFm = "ImageScanFindings";
const _ISS = "ImageScanStatus";
const _ISSL = "ImageSigningStatusList";
const _ISSm = "ImageSigningStatus";
const _ITAEE = "ImageTagAlreadyExistsException";
const _ITMEF = "ImageTagMutabilityExclusionFilter";
const _ITMEFm = "ImageTagMutabilityExclusionFilters";
const _ITPE = "InvalidTagParameterException";
const _K = "Key";
const _KE = "KmsException";
const _L = "Layer";
const _LAEE = "LayerAlreadyExistsException";
const _LEE = "LimitExceededException";
const _LF = "LayerFailure";
const _LFL = "LayerFailureList";
const _LI = "ListImages";
const _LIE = "LayerInaccessibleException";
const _LIF = "ListImagesFilter";
const _LIR = "ListImagesRequest";
const _LIRF = "ListImageReferrersFilter";
const _LIRR = "ListImageReferrersRequest";
const _LIRRi = "ListImageReferrersResponse";
const _LIRi = "ListImagesResponse";
const _LIRis = "ListImageReferrers";
const _LL = "LayerList";
const _LNFE = "LayersNotFoundException";
const _LPNFE = "LifecyclePolicyNotFoundException";
const _LPPF = "LifecyclePolicyPreviewFilter";
const _LPPIPE = "LifecyclePolicyPreviewInProgressException";
const _LPPNFE = "LifecyclePolicyPreviewNotFoundException";
const _LPPR = "LifecyclePolicyPreviewResult";
const _LPPRL = "LifecyclePolicyPreviewResultList";
const _LPPS = "LifecyclePolicyPreviewSummary";
const _LPRA = "LifecyclePolicyRuleAction";
const _LPTSE = "LayerPartTooSmallException";
const _LPTUE = "ListPullTimeUpdateExclusions";
const _LPTUER = "ListPullTimeUpdateExclusionsRequest";
const _LPTUERi = "ListPullTimeUpdateExclusionsResponse";
const _LTFR = "ListTagsForResource";
const _LTFRR = "ListTagsForResourceRequest";
const _LTFRRi = "ListTagsForResourceResponse";
const _PAS = "PutAccountSetting";
const _PASR = "PutAccountSettingRequest";
const _PASRu = "PutAccountSettingResponse";
const _PI = "PutImage";
const _PIR = "PutImageRequest";
const _PIRu = "PutImageResponse";
const _PISC = "PutImageScanningConfiguration";
const _PISCR = "PutImageScanningConfigurationRequest";
const _PISCRu = "PutImageScanningConfigurationResponse";
const _PITM = "PutImageTagMutability";
const _PITMR = "PutImageTagMutabilityRequest";
const _PITMRu = "PutImageTagMutabilityResponse";
const _PLP = "PutLifecyclePolicy";
const _PLPR = "PutLifecyclePolicyRequest";
const _PLPRu = "PutLifecyclePolicyResponse";
const _PRC = "PutReplicationConfiguration";
const _PRCR = "PutReplicationConfigurationRequest";
const _PRCRu = "PutReplicationConfigurationResponse";
const _PRP = "PutRegistryPolicy";
const _PRPR = "PutRegistryPolicyRequest";
const _PRPRu = "PutRegistryPolicyResponse";
const _PRSC = "PutRegistryScanningConfiguration";
const _PRSCR = "PutRegistryScanningConfigurationRequest";
const _PRSCRu = "PutRegistryScanningConfigurationResponse";
const _PSC = "PutSigningConfiguration";
const _PSCR = "PutSigningConfigurationRequest";
const _PSCRu = "PutSigningConfigurationResponse";
const _PTCR = "PullThroughCacheRule";
const _PTCRAEE = "PullThroughCacheRuleAlreadyExistsException";
const _PTCRL = "PullThroughCacheRuleList";
const _PTCRNFE = "PullThroughCacheRuleNotFoundException";
const _PVD = "PackageVulnerabilityDetails";
const _R = "Recommendation";
const _RAEE = "RepositoryAlreadyExistsException";
const _RC = "ReplicationConfiguration";
const _RCT = "RepositoryCreationTemplate";
const _RCTL = "RepositoryCreationTemplateList";
const _RD = "ReplicationDestination";
const _RDL = "ReplicationDestinationList";
const _RDe = "ResourceDetails";
const _RF = "RepositoryFilter";
const _RFL = "RepositoryFilterList";
const _RINFE = "ReferencedImagesNotFoundException";
const _RL = "RepositoryList";
const _RLe = "ResourceList";
const _RNEE = "RepositoryNotEmptyException";
const _RNFE = "RepositoryNotFoundException";
const _RPNFE = "RegistryPolicyNotFoundException";
const _RPNFEe = "RepositoryPolicyNotFoundException";
const _RPTUE = "RegisterPullTimeUpdateExclusion";
const _RPTUER = "RegisterPullTimeUpdateExclusionRequest";
const _RPTUERe = "RegisterPullTimeUpdateExclusionResponse";
const _RR = "ReplicationRule";
const _RRL = "ReplicationRuleList";
const _RSC = "RegistryScanningConfiguration";
const _RSCF = "RepositoryScanningConfigurationFailure";
const _RSCFL = "RepositoryScanningConfigurationFailureList";
const _RSCL = "RepositoryScanningConfigurationList";
const _RSCe = "RepositoryScanningConfiguration";
const _RSR = "RegistryScanningRule";
const _RSRL = "RegistryScanningRuleList";
const _Re = "Remediation";
const _Rep = "Repository";
const _Res = "Resource";
const _SC = "SigningConfiguration";
const _SCNFE = "SigningConfigurationNotFoundException";
const _SD = "ScoreDetails";
const _SE = "ServerException";
const _SI = "SubjectIdentifier";
const _SIS = "StartImageScan";
const _SISR = "StartImageScanRequest";
const _SISRt = "StartImageScanResponse";
const _SLPP = "StartLifecyclePolicyPreview";
const _SLPPR = "StartLifecyclePolicyPreviewRequest";
const _SLPPRt = "StartLifecyclePolicyPreviewResponse";
const _SNFE = "ScanNotFoundException";
const _SNFEe = "SecretNotFoundException";
const _SR = "SigningRule";
const _SRF = "ScanningRepositoryFilter";
const _SRFL = "ScanningRepositoryFilterList";
const _SRFLi = "SigningRepositoryFilterList";
const _SRFi = "SigningRepositoryFilter";
const _SRL = "SigningRuleList";
const _SRP = "SetRepositoryPolicy";
const _SRPR = "SetRepositoryPolicyRequest";
const _SRPRe = "SetRepositoryPolicyResponse";
const _T = "Tag";
const _TAEE = "TemplateAlreadyExistsException";
const _TITC = "TransitioningImageTotalCount";
const _TITCr = "TransitioningImageTotalCounts";
const _TL = "TagList";
const _TMTE = "TooManyTagsException";
const _TNFE = "TemplateNotFoundException";
const _TR = "TagResource";
const _TRR = "TagResourceRequest";
const _TRRa = "TagResourceResponse";
const _UISC = "UpdateImageStorageClass";
const _UISCR = "UpdateImageStorageClassRequest";
const _UISCRp = "UpdateImageStorageClassResponse";
const _UITE = "UnsupportedImageTypeException";
const _ULP = "UploadLayerPart";
const _ULPR = "UploadLayerPartRequest";
const _ULPRp = "UploadLayerPartResponse";
const _UNFE = "UploadNotFoundException";
const _UPTCR = "UpdatePullThroughCacheRule";
const _UPTCRR = "UpdatePullThroughCacheRuleRequest";
const _UPTCRRp = "UpdatePullThroughCacheRuleResponse";
const _UR = "UntagResource";
const _URCT = "UpdateRepositoryCreationTemplate";
const _URCTR = "UpdateRepositoryCreationTemplateRequest";
const _URCTRp = "UpdateRepositoryCreationTemplateResponse";
const _URR = "UntagResourceRequest";
const _URRn = "UntagResourceResponse";
const _UTASE = "UnableToAccessSecretException";
const _UTDSVE = "UnableToDecryptSecretValueException";
const _UTGUIE = "UnableToGetUpstreamImageException";
const _UTGULE = "UnableToGetUpstreamLayerException";
const _UTLUIRE = "UnableToListUpstreamImageReferrersException";
const _UURE = "UnsupportedUpstreamRegistryException";
const _V = "Value";
const _VE = "ValidationException";
const _VP = "VulnerablePackage";
const _VPL = "VulnerablePackagesList";
const _VPTCR = "ValidatePullThroughCacheRule";
const _VPTCRR = "ValidatePullThroughCacheRuleRequest";
const _VPTCRRa = "ValidatePullThroughCacheRuleResponse";
const _a = "architecture";
const _aAI = "awsAccountId";
const _aD = "authorizationData";
const _aECI = "awsEcrContainerImage";
const _aF = "appliedFor";
const _aMT = "acceptedMediaTypes";
const _aMTr = "artifactMediaType";
const _aRP = "appliedRulePriority";
const _aS = "artifactStatus";
const _aSF = "appliedScanFilters";
const _aT = "authorizationToken";
const _aTr = "artifactType";
const _aTrt = "artifactTypes";
const _ac = "action";
const _ad = "adjustments";
const _an = "annotations";
const _ar = "arch";
const _at = "attributes";
const _au = "author";
const _bS = "baseScore";
const _c = "client";
const _cA = "credentialArn";
const _cAr = "createdAt";
const _cRA = "customRoleArn";
const _cv = "cvss";
const _d = "description";
const _dU = "downloadUrl";
const _de = "destinations";
const _det = "details";
const _di = "digest";
const _e = "error";
const _eA = "expiresAt";
const _eAx = "exploitAvailable";
const _eC = "encryptionConfiguration";
const _eF = "enhancedFindings";
const _eITC = "expiringImageTotalCount";
const _eRP = "ecrRepositoryPrefix";
const _eRPc = "ecrRepositoryPrefixes";
const _eT = "encryptionType";
const _ep = "epoch";
const _f = "failures";
const _fA = "findingArn";
const _fAi = "fixAvailable";
const _fC = "failureCode";
const _fIV = "fixedInVersion";
const _fOA = "firstObservedAt";
const _fP = "filePath";
const _fR = "failureReason";
const _fSC = "findingSeverityCounts";
const _fT = "filterType";
const _fa = "failure";
const _fi = "filter";
const _fin = "findings";
const _fo = "force";
const _hE = "httpError";
const _i = "images";
const _iD = "imageDetails";
const _iDm = "imageDigest";
const _iH = "imageHash";
const _iI = "imageIds";
const _iIm = "imageId";
const _iM = "imageManifest";
const _iMMT = "imageManifestMediaType";
const _iPA = "imagePushedAt";
const _iS = "imageStatus";
const _iSC = "imageScanningConfiguration";
const _iSCA = "imageScanCompletedAt";
const _iSF = "imageScanFindings";
const _iSFS = "imageScanFindingsSummary";
const _iSIB = "imageSizeInBytes";
const _iSS = "imageScanStatus";
const _iT = "imageTags";
const _iTC = "imageTotalCount";
const _iTM = "imageTagMutability";
const _iTMEF = "imageTagMutabilityExclusionFilters";
const _iTm = "imageTag";
const _iUC = "inUseCount";
const _iV = "isValid";
const _id = "id";
const _im = "image";
const _k = "key";
const _kE = "kmsError";
const _kK = "kmsKey";
const _l = "layers";
const _lA = "layerAvailability";
const _lAA = "lastArchivedAt";
const _lAAa = "lastActivatedAt";
const _lBR = "lastByteReceived";
const _lD = "layerDigests";
const _lDa = "layerDigest";
const _lEA = "lastEvaluatedAt";
const _lIUA = "lastInUseAt";
const _lOA = "lastObservedAt";
const _lP = "lifecyclePolicy";
const _lPB = "layerPartBlob";
const _lPT = "lifecyclePolicyText";
const _lRPT = "lastRecordedPullTime";
const _lS = "layerSize";
const _lVBR = "lastValidByteReceived";
const _m = "message";
const _mR = "maxResults";
const _mT = "mediaType";
const _me = "metric";
const _n = "name";
const _nT = "nextToken";
const _p = "platform";
const _pA = "pushedAt";
const _pAr = "principalArn";
const _pE = "proxyEndpoint";
const _pFB = "partFirstByte";
const _pLB = "partLastByte";
const _pM = "packageManager";
const _pR = "previewResults";
const _pS = "partSize";
const _pT = "policyText";
const _pTCR = "pullThroughCacheRules";
const _pTUE = "pullTimeUpdateExclusions";
const _pVD = "packageVulnerabilityDetails";
const _pr = "prefix";
const _pre = "prefixes";
const _r = "registry";
const _rA = "resourceArn";
const _rAe = "repositoryArn";
const _rC = "replicationConfiguration";
const _rCT = "repositoryCreationTemplate";
const _rCTe = "repositoryCreationTemplates";
const _rF = "repositoryFilters";
const _rI = "registryId";
const _rIe = "registryIds";
const _rN = "repositoryName";
const _rNe = "repositoryNames";
const _rP = "repositoryPolicy";
const _rS = "replicationStatuses";
const _rSC = "registryScanningConfiguration";
const _rT = "resourceTags";
const _rU = "referenceUrls";
const _rUe = "repositoryUri";
const _rV = "relatedVulnerabilities";
const _re = "repository";
const _rea = "reason";
const _rec = "recommendation";
const _ref = "referrers";
const _reg = "region";
const _rel = "release";
const _rem = "remediation";
const _rep = "repositories";
const _res = "resources";
const _ru = "rules";
const _s = "smithy.ts.sdk.synthetic.com.amazonaws.ecr";
const _sC = "scanningConfigurations";
const _sCc = "scanningConfiguration";
const _sCi = "signingConfiguration";
const _sCt = "storageClass";
const _sD = "scoreDetails";
const _sF = "scanFrequency";
const _sI = "subjectId";
const _sLH = "sourceLayerHash";
const _sMD = "subjectManifestDigest";
const _sOP = "scanOnPush";
const _sPA = "signingProfileArn";
const _sS = "scoreSource";
const _sSi = "signingStatuses";
const _sT = "scanType";
const _sU = "sourceUrl";
const _sV = "scoringVector";
const _sc = "score";
const _se = "server";
const _sev = "severity";
const _si = "size";
const _so = "source";
const _st = "status";
const _su = "summary";
const _t = "tags";
const _tITC = "transitioningImageTotalCounts";
const _tK = "tagKeys";
const _tS = "tagStatus";
const _tSC = "targetStorageClass";
const _te = "text";
const _ti = "title";
const _ty = "type";
const _u = "uri";
const _uA = "updatedAt";
const _uI = "uploadId";
const _uR = "upstreamRegistry";
const _uRP = "upstreamRepositoryPrefix";
const _uRU = "upstreamRegistryUrl";
const _ur = "url";
const _v = "value";
const _vCA = "vendorCreatedAt";
const _vI = "vulnerabilityId";
const _vP = "vulnerablePackages";
const _vS = "vendorSeverity";
const _vSUA = "vulnerabilitySourceUpdatedAt";
const _vUA = "vendorUpdatedAt";
const _ve = "version";
const n0 = "com.amazonaws.ecr";
const { TypeRegistry } = require("@smithy/core/schema");
const { ECRServiceException } = require("../models/ECRServiceException");
const { BlockedByOrganizationPolicyException, EmptyUploadException, ExclusionAlreadyExistsException, ExclusionNotFoundException, ImageAlreadyExistsException, ImageArchivedException, ImageDigestDoesNotMatchException, ImageNotFoundException, ImageStorageClassUpdateNotSupportedException, ImageTagAlreadyExistsException, InvalidLayerException, InvalidLayerPartException, InvalidParameterException, InvalidTagParameterException, KmsException, LayerAlreadyExistsException, LayerInaccessibleException, LayerPartTooSmallException, LayersNotFoundException, LifecyclePolicyNotFoundException, LifecyclePolicyPreviewInProgressException, LifecyclePolicyPreviewNotFoundException, LimitExceededException, PullThroughCacheRuleAlreadyExistsException, PullThroughCacheRuleNotFoundException, ReferencedImagesNotFoundException, RegistryPolicyNotFoundException, RepositoryAlreadyExistsException, RepositoryNotEmptyException, RepositoryNotFoundException, RepositoryPolicyNotFoundException, ScanNotFoundException, SecretNotFoundException, ServerException, SigningConfigurationNotFoundException, TemplateAlreadyExistsException, TemplateNotFoundException, TooManyTagsException, UnableToAccessSecretException, UnableToDecryptSecretValueException, UnableToGetUpstreamImageException, UnableToGetUpstreamLayerException, UnableToListUpstreamImageReferrersException, UnsupportedImageTypeException, UnsupportedUpstreamRegistryException, UploadNotFoundException, ValidationException } = require("../models/errors");
const _s_registry = TypeRegistry.for(_s);
const ECRServiceException$ = [-3, _s, "ECRServiceException", 0, [], []];
exports.ECRServiceException$ = ECRServiceException$;
_s_registry.registerError(ECRServiceException$, ECRServiceException);
const n0_registry = TypeRegistry.for(n0);
const BlockedByOrganizationPolicyException$ = [-3, n0, _BBOPE,
    { [_e]: _c },
    [_m],
    [0]
];
exports.BlockedByOrganizationPolicyException$ = BlockedByOrganizationPolicyException$;
n0_registry.registerError(BlockedByOrganizationPolicyException$, BlockedByOrganizationPolicyException);
const EmptyUploadException$ = [-3, n0, _EUE,
    { [_e]: _c },
    [_m],
    [0]
];
exports.EmptyUploadException$ = EmptyUploadException$;
n0_registry.registerError(EmptyUploadException$, EmptyUploadException);
const ExclusionAlreadyExistsException$ = [-3, n0, _EAEE,
    { [_e]: _c },
    [_m],
    [0]
];
exports.ExclusionAlreadyExistsException$ = ExclusionAlreadyExistsException$;
n0_registry.registerError(ExclusionAlreadyExistsException$, ExclusionAlreadyExistsException);
const ExclusionNotFoundException$ = [-3, n0, _ENFE,
    { [_e]: _c },
    [_m],
    [0]
];
exports.ExclusionNotFoundException$ = ExclusionNotFoundException$;
n0_registry.registerError(ExclusionNotFoundException$, ExclusionNotFoundException);
const ImageAlreadyExistsException$ = [-3, n0, _IAEE,
    { [_e]: _c },
    [_m],
    [0]
];
exports.ImageAlreadyExistsException$ = ImageAlreadyExistsException$;
n0_registry.registerError(ImageAlreadyExistsException$, ImageAlreadyExistsException);
const ImageArchivedException$ = [-3, n0, _IAE,
    { [_e]: _c },
    [_m],
    [0]
];
exports.ImageArchivedException$ = ImageArchivedException$;
n0_registry.registerError(ImageArchivedException$, ImageArchivedException);
const ImageDigestDoesNotMatchException$ = [-3, n0, _IDDNME,
    { [_e]: _c },
    [_m],
    [0]
];
exports.ImageDigestDoesNotMatchException$ = ImageDigestDoesNotMatchException$;
n0_registry.registerError(ImageDigestDoesNotMatchException$, ImageDigestDoesNotMatchException);
const ImageNotFoundException$ = [-3, n0, _INFE,
    { [_e]: _c },
    [_m],
    [0]
];
exports.ImageNotFoundException$ = ImageNotFoundException$;
n0_registry.registerError(ImageNotFoundException$, ImageNotFoundException);
const ImageStorageClassUpdateNotSupportedException$ = [-3, n0, _ISCUNSE,
    { [_e]: _c },
    [_m],
    [0]
];
exports.ImageStorageClassUpdateNotSupportedException$ = ImageStorageClassUpdateNotSupportedException$;
n0_registry.registerError(ImageStorageClassUpdateNotSupportedException$, ImageStorageClassUpdateNotSupportedException);
const ImageTagAlreadyExistsException$ = [-3, n0, _ITAEE,
    { [_e]: _c },
    [_m],
    [0]
];
exports.ImageTagAlreadyExistsException$ = ImageTagAlreadyExistsException$;
n0_registry.registerError(ImageTagAlreadyExistsException$, ImageTagAlreadyExistsException);
const InvalidLayerException$ = [-3, n0, _ILE,
    { [_e]: _c },
    [_m],
    [0]
];
exports.InvalidLayerException$ = InvalidLayerException$;
n0_registry.registerError(InvalidLayerException$, InvalidLayerException);
const InvalidLayerPartException$ = [-3, n0, _ILPE,
    { [_e]: _c },
    [_rI, _rN, _uI, _lVBR, _m],
    [0, 0, 0, 1, 0]
];
exports.InvalidLayerPartException$ = InvalidLayerPartException$;
n0_registry.registerError(InvalidLayerPartException$, InvalidLayerPartException);
const InvalidParameterException$ = [-3, n0, _IPE,
    { [_e]: _c },
    [_m],
    [0]
];
exports.InvalidParameterException$ = InvalidParameterException$;
n0_registry.registerError(InvalidParameterException$, InvalidParameterException);
const InvalidTagParameterException$ = [-3, n0, _ITPE,
    { [_e]: _c },
    [_m],
    [0]
];
exports.InvalidTagParameterException$ = InvalidTagParameterException$;
n0_registry.registerError(InvalidTagParameterException$, InvalidTagParameterException);
const KmsException$ = [-3, n0, _KE,
    { [_e]: _c },
    [_m, _kE],
    [0, 0]
];
exports.KmsException$ = KmsException$;
n0_registry.registerError(KmsException$, KmsException);
const LayerAlreadyExistsException$ = [-3, n0, _LAEE,
    { [_e]: _c },
    [_m],
    [0]
];
exports.LayerAlreadyExistsException$ = LayerAlreadyExistsException$;
n0_registry.registerError(LayerAlreadyExistsException$, LayerAlreadyExistsException);
const LayerInaccessibleException$ = [-3, n0, _LIE,
    { [_e]: _c },
    [_m],
    [0]
];
exports.LayerInaccessibleException$ = LayerInaccessibleException$;
n0_registry.registerError(LayerInaccessibleException$, LayerInaccessibleException);
const LayerPartTooSmallException$ = [-3, n0, _LPTSE,
    { [_e]: _c },
    [_m],
    [0]
];
exports.LayerPartTooSmallException$ = LayerPartTooSmallException$;
n0_registry.registerError(LayerPartTooSmallException$, LayerPartTooSmallException);
const LayersNotFoundException$ = [-3, n0, _LNFE,
    { [_e]: _c },
    [_m],
    [0]
];
exports.LayersNotFoundException$ = LayersNotFoundException$;
n0_registry.registerError(LayersNotFoundException$, LayersNotFoundException);
const LifecyclePolicyNotFoundException$ = [-3, n0, _LPNFE,
    { [_e]: _c },
    [_m],
    [0]
];
exports.LifecyclePolicyNotFoundException$ = LifecyclePolicyNotFoundException$;
n0_registry.registerError(LifecyclePolicyNotFoundException$, LifecyclePolicyNotFoundException);
const LifecyclePolicyPreviewInProgressException$ = [-3, n0, _LPPIPE,
    { [_e]: _c },
    [_m],
    [0]
];
exports.LifecyclePolicyPreviewInProgressException$ = LifecyclePolicyPreviewInProgressException$;
n0_registry.registerError(LifecyclePolicyPreviewInProgressException$, LifecyclePolicyPreviewInProgressException);
const LifecyclePolicyPreviewNotFoundException$ = [-3, n0, _LPPNFE,
    { [_e]: _c },
    [_m],
    [0]
];
exports.LifecyclePolicyPreviewNotFoundException$ = LifecyclePolicyPreviewNotFoundException$;
n0_registry.registerError(LifecyclePolicyPreviewNotFoundException$, LifecyclePolicyPreviewNotFoundException);
const LimitExceededException$ = [-3, n0, _LEE,
    { [_e]: _c },
    [_m],
    [0]
];
exports.LimitExceededException$ = LimitExceededException$;
n0_registry.registerError(LimitExceededException$, LimitExceededException);
const PullThroughCacheRuleAlreadyExistsException$ = [-3, n0, _PTCRAEE,
    { [_e]: _c },
    [_m],
    [0]
];
exports.PullThroughCacheRuleAlreadyExistsException$ = PullThroughCacheRuleAlreadyExistsException$;
n0_registry.registerError(PullThroughCacheRuleAlreadyExistsException$, PullThroughCacheRuleAlreadyExistsException);
const PullThroughCacheRuleNotFoundException$ = [-3, n0, _PTCRNFE,
    { [_e]: _c },
    [_m],
    [0]
];
exports.PullThroughCacheRuleNotFoundException$ = PullThroughCacheRuleNotFoundException$;
n0_registry.registerError(PullThroughCacheRuleNotFoundException$, PullThroughCacheRuleNotFoundException);
const ReferencedImagesNotFoundException$ = [-3, n0, _RINFE,
    { [_e]: _c },
    [_m],
    [0]
];
exports.ReferencedImagesNotFoundException$ = ReferencedImagesNotFoundException$;
n0_registry.registerError(ReferencedImagesNotFoundException$, ReferencedImagesNotFoundException);
const RegistryPolicyNotFoundException$ = [-3, n0, _RPNFE,
    { [_e]: _c },
    [_m],
    [0]
];
exports.RegistryPolicyNotFoundException$ = RegistryPolicyNotFoundException$;
n0_registry.registerError(RegistryPolicyNotFoundException$, RegistryPolicyNotFoundException);
const RepositoryAlreadyExistsException$ = [-3, n0, _RAEE,
    { [_e]: _c },
    [_m],
    [0]
];
exports.RepositoryAlreadyExistsException$ = RepositoryAlreadyExistsException$;
n0_registry.registerError(RepositoryAlreadyExistsException$, RepositoryAlreadyExistsException);
const RepositoryNotEmptyException$ = [-3, n0, _RNEE,
    { [_e]: _c },
    [_m],
    [0]
];
exports.RepositoryNotEmptyException$ = RepositoryNotEmptyException$;
n0_registry.registerError(RepositoryNotEmptyException$, RepositoryNotEmptyException);
const RepositoryNotFoundException$ = [-3, n0, _RNFE,
    { [_e]: _c },
    [_m],
    [0]
];
exports.RepositoryNotFoundException$ = RepositoryNotFoundException$;
n0_registry.registerError(RepositoryNotFoundException$, RepositoryNotFoundException);
const RepositoryPolicyNotFoundException$ = [-3, n0, _RPNFEe,
    { [_e]: _c },
    [_m],
    [0]
];
exports.RepositoryPolicyNotFoundException$ = RepositoryPolicyNotFoundException$;
n0_registry.registerError(RepositoryPolicyNotFoundException$, RepositoryPolicyNotFoundException);
const ScanNotFoundException$ = [-3, n0, _SNFE,
    { [_e]: _c },
    [_m],
    [0]
];
exports.ScanNotFoundException$ = ScanNotFoundException$;
n0_registry.registerError(ScanNotFoundException$, ScanNotFoundException);
const SecretNotFoundException$ = [-3, n0, _SNFEe,
    { [_e]: _c },
    [_m],
    [0]
];
exports.SecretNotFoundException$ = SecretNotFoundException$;
n0_registry.registerError(SecretNotFoundException$, SecretNotFoundException);
const ServerException$ = [-3, n0, _SE,
    { [_e]: _se },
    [_m],
    [0]
];
exports.ServerException$ = ServerException$;
n0_registry.registerError(ServerException$, ServerException);
const SigningConfigurationNotFoundException$ = [-3, n0, _SCNFE,
    { [_e]: _c },
    [_m],
    [0]
];
exports.SigningConfigurationNotFoundException$ = SigningConfigurationNotFoundException$;
n0_registry.registerError(SigningConfigurationNotFoundException$, SigningConfigurationNotFoundException);
const TemplateAlreadyExistsException$ = [-3, n0, _TAEE,
    { [_e]: _c },
    [_m],
    [0]
];
exports.TemplateAlreadyExistsException$ = TemplateAlreadyExistsException$;
n0_registry.registerError(TemplateAlreadyExistsException$, TemplateAlreadyExistsException);
const TemplateNotFoundException$ = [-3, n0, _TNFE,
    { [_e]: _c },
    [_m],
    [0]
];
exports.TemplateNotFoundException$ = TemplateNotFoundException$;
n0_registry.registerError(TemplateNotFoundException$, TemplateNotFoundException);
const TooManyTagsException$ = [-3, n0, _TMTE,
    { [_e]: _c },
    [_m],
    [0]
];
exports.TooManyTagsException$ = TooManyTagsException$;
n0_registry.registerError(TooManyTagsException$, TooManyTagsException);
const UnableToAccessSecretException$ = [-3, n0, _UTASE,
    { [_e]: _c },
    [_m],
    [0]
];
exports.UnableToAccessSecretException$ = UnableToAccessSecretException$;
n0_registry.registerError(UnableToAccessSecretException$, UnableToAccessSecretException);
const UnableToDecryptSecretValueException$ = [-3, n0, _UTDSVE,
    { [_e]: _c },
    [_m],
    [0]
];
exports.UnableToDecryptSecretValueException$ = UnableToDecryptSecretValueException$;
n0_registry.registerError(UnableToDecryptSecretValueException$, UnableToDecryptSecretValueException);
const UnableToGetUpstreamImageException$ = [-3, n0, _UTGUIE,
    { [_e]: _c },
    [_m],
    [0]
];
exports.UnableToGetUpstreamImageException$ = UnableToGetUpstreamImageException$;
n0_registry.registerError(UnableToGetUpstreamImageException$, UnableToGetUpstreamImageException);
const UnableToGetUpstreamLayerException$ = [-3, n0, _UTGULE,
    { [_e]: _c },
    [_m],
    [0]
];
exports.UnableToGetUpstreamLayerException$ = UnableToGetUpstreamLayerException$;
n0_registry.registerError(UnableToGetUpstreamLayerException$, UnableToGetUpstreamLayerException);
const UnableToListUpstreamImageReferrersException$ = [-3, n0, _UTLUIRE,
    { [_e]: _c },
    [_m],
    [0]
];
exports.UnableToListUpstreamImageReferrersException$ = UnableToListUpstreamImageReferrersException$;
n0_registry.registerError(UnableToListUpstreamImageReferrersException$, UnableToListUpstreamImageReferrersException);
const UnsupportedImageTypeException$ = [-3, n0, _UITE,
    { [_e]: _c },
    [_m],
    [0]
];
exports.UnsupportedImageTypeException$ = UnsupportedImageTypeException$;
n0_registry.registerError(UnsupportedImageTypeException$, UnsupportedImageTypeException);
const UnsupportedUpstreamRegistryException$ = [-3, n0, _UURE,
    { [_e]: _c },
    [_m],
    [0]
];
exports.UnsupportedUpstreamRegistryException$ = UnsupportedUpstreamRegistryException$;
n0_registry.registerError(UnsupportedUpstreamRegistryException$, UnsupportedUpstreamRegistryException);
const UploadNotFoundException$ = [-3, n0, _UNFE,
    { [_e]: _c },
    [_m],
    [0]
];
exports.UploadNotFoundException$ = UploadNotFoundException$;
n0_registry.registerError(UploadNotFoundException$, UploadNotFoundException);
const ValidationException$ = [-3, n0, _VE,
    { [_e]: _c, [_hE]: 400 },
    [_m],
    [0]
];
exports.ValidationException$ = ValidationException$;
n0_registry.registerError(ValidationException$, ValidationException);
exports.errorTypeRegistries = [
    _s_registry,
    n0_registry,
];
const Attribute$ = [3, n0, _A,
    0,
    [_k, _v],
    [0, 0], 1
];
exports.Attribute$ = Attribute$;
const AuthorizationData$ = [3, n0, _AD,
    0,
    [_aT, _eA, _pE],
    [0, 4, 0]
];
exports.AuthorizationData$ = AuthorizationData$;
const AwsEcrContainerImageDetails$ = [3, n0, _AECID,
    0,
    [_a, _au, _iH, _iT, _p, _pA, _lIUA, _iUC, _r, _rN],
    [0, 0, 0, 64 | 0, 0, 4, 4, 1, 0, 0]
];
exports.AwsEcrContainerImageDetails$ = AwsEcrContainerImageDetails$;
const BatchCheckLayerAvailabilityRequest$ = [3, n0, _BCLAR,
    0,
    [_rN, _lD, _rI],
    [0, 64 | 0, 0], 2
];
exports.BatchCheckLayerAvailabilityRequest$ = BatchCheckLayerAvailabilityRequest$;
const BatchCheckLayerAvailabilityResponse$ = [3, n0, _BCLARa,
    0,
    [_l, _f],
    [() => LayerList, () => LayerFailureList]
];
exports.BatchCheckLayerAvailabilityResponse$ = BatchCheckLayerAvailabilityResponse$;
const BatchDeleteImageRequest$ = [3, n0, _BDIR,
    0,
    [_rN, _iI, _rI],
    [0, () => ImageIdentifierList, 0], 2
];
exports.BatchDeleteImageRequest$ = BatchDeleteImageRequest$;
const BatchDeleteImageResponse$ = [3, n0, _BDIRa,
    0,
    [_iI, _f],
    [() => ImageIdentifierList, () => ImageFailureList]
];
exports.BatchDeleteImageResponse$ = BatchDeleteImageResponse$;
const BatchGetImageRequest$ = [3, n0, _BGIR,
    0,
    [_rN, _iI, _rI, _aMT],
    [0, () => ImageIdentifierList, 0, 64 | 0], 2
];
exports.BatchGetImageRequest$ = BatchGetImageRequest$;
const BatchGetImageResponse$ = [3, n0, _BGIRa,
    0,
    [_i, _f],
    [() => ImageList, () => ImageFailureList]
];
exports.BatchGetImageResponse$ = BatchGetImageResponse$;
const BatchGetRepositoryScanningConfigurationRequest$ = [3, n0, _BGRSCR,
    0,
    [_rNe],
    [64 | 0], 1
];
exports.BatchGetRepositoryScanningConfigurationRequest$ = BatchGetRepositoryScanningConfigurationRequest$;
const BatchGetRepositoryScanningConfigurationResponse$ = [3, n0, _BGRSCRa,
    0,
    [_sC, _f],
    [() => RepositoryScanningConfigurationList, () => RepositoryScanningConfigurationFailureList]
];
exports.BatchGetRepositoryScanningConfigurationResponse$ = BatchGetRepositoryScanningConfigurationResponse$;
const CompleteLayerUploadRequest$ = [3, n0, _CLUR,
    0,
    [_rN, _uI, _lD, _rI],
    [0, 0, 64 | 0, 0], 3
];
exports.CompleteLayerUploadRequest$ = CompleteLayerUploadRequest$;
const CompleteLayerUploadResponse$ = [3, n0, _CLURo,
    0,
    [_rI, _rN, _uI, _lDa],
    [0, 0, 0, 0]
];
exports.CompleteLayerUploadResponse$ = CompleteLayerUploadResponse$;
const CreatePullThroughCacheRuleRequest$ = [3, n0, _CPTCRR,
    0,
    [_eRP, _uRU, _rI, _uR, _cA, _cRA, _uRP],
    [0, 0, 0, 0, 0, 0, 0], 2
];
exports.CreatePullThroughCacheRuleRequest$ = CreatePullThroughCacheRuleRequest$;
const CreatePullThroughCacheRuleResponse$ = [3, n0, _CPTCRRr,
    0,
    [_eRP, _uRU, _cAr, _rI, _uR, _cA, _cRA, _uRP],
    [0, 0, 4, 0, 0, 0, 0, 0]
];
exports.CreatePullThroughCacheRuleResponse$ = CreatePullThroughCacheRuleResponse$;
const CreateRepositoryCreationTemplateRequest$ = [3, n0, _CRCTR,
    0,
    [_pr, _aF, _d, _eC, _rT, _iTM, _iTMEF, _rP, _lP, _cRA],
    [0, 64 | 0, 0, () => EncryptionConfigurationForRepositoryCreationTemplate$, () => TagList, 0, () => ImageTagMutabilityExclusionFilters, 0, 0, 0], 2
];
exports.CreateRepositoryCreationTemplateRequest$ = CreateRepositoryCreationTemplateRequest$;
const CreateRepositoryCreationTemplateResponse$ = [3, n0, _CRCTRr,
    0,
    [_rI, _rCT],
    [0, () => RepositoryCreationTemplate$]
];
exports.CreateRepositoryCreationTemplateResponse$ = CreateRepositoryCreationTemplateResponse$;
const CreateRepositoryRequest$ = [3, n0, _CRR,
    0,
    [_rN, _rI, _t, _iTM, _iTMEF, _iSC, _eC],
    [0, 0, () => TagList, 0, () => ImageTagMutabilityExclusionFilters, () => ImageScanningConfiguration$, () => EncryptionConfiguration$], 1
];
exports.CreateRepositoryRequest$ = CreateRepositoryRequest$;
const CreateRepositoryResponse$ = [3, n0, _CRRr,
    0,
    [_re],
    [() => Repository$]
];
exports.CreateRepositoryResponse$ = CreateRepositoryResponse$;
const CvssScore$ = [3, n0, _CS,
    0,
    [_bS, _sV, _so, _ve],
    [1, 0, 0, 0]
];
exports.CvssScore$ = CvssScore$;
const CvssScoreAdjustment$ = [3, n0, _CSA,
    0,
    [_me, _rea],
    [0, 0]
];
exports.CvssScoreAdjustment$ = CvssScoreAdjustment$;
const CvssScoreDetails$ = [3, n0, _CSD,
    0,
    [_ad, _sc, _sS, _sV, _ve],
    [() => CvssScoreAdjustmentList, 1, 0, 0, 0]
];
exports.CvssScoreDetails$ = CvssScoreDetails$;
const DeleteLifecyclePolicyRequest$ = [3, n0, _DLPR,
    0,
    [_rN, _rI],
    [0, 0], 1
];
exports.DeleteLifecyclePolicyRequest$ = DeleteLifecyclePolicyRequest$;
const DeleteLifecyclePolicyResponse$ = [3, n0, _DLPRe,
    0,
    [_rI, _rN, _lPT, _lEA],
    [0, 0, 0, 4]
];
exports.DeleteLifecyclePolicyResponse$ = DeleteLifecyclePolicyResponse$;
const DeletePullThroughCacheRuleRequest$ = [3, n0, _DPTCRR,
    0,
    [_eRP, _rI],
    [0, 0], 1
];
exports.DeletePullThroughCacheRuleRequest$ = DeletePullThroughCacheRuleRequest$;
const DeletePullThroughCacheRuleResponse$ = [3, n0, _DPTCRRe,
    0,
    [_eRP, _uRU, _cAr, _rI, _cA, _cRA, _uRP],
    [0, 0, 4, 0, 0, 0, 0]
];
exports.DeletePullThroughCacheRuleResponse$ = DeletePullThroughCacheRuleResponse$;
const DeleteRegistryPolicyRequest$ = [3, n0, _DRPR,
    0,
    [],
    []
];
exports.DeleteRegistryPolicyRequest$ = DeleteRegistryPolicyRequest$;
const DeleteRegistryPolicyResponse$ = [3, n0, _DRPRe,
    0,
    [_rI, _pT],
    [0, 0]
];
exports.DeleteRegistryPolicyResponse$ = DeleteRegistryPolicyResponse$;
const DeleteRepositoryCreationTemplateRequest$ = [3, n0, _DRCTR,
    0,
    [_pr],
    [0], 1
];
exports.DeleteRepositoryCreationTemplateRequest$ = DeleteRepositoryCreationTemplateRequest$;
const DeleteRepositoryCreationTemplateResponse$ = [3, n0, _DRCTRe,
    0,
    [_rI, _rCT],
    [0, () => RepositoryCreationTemplate$]
];
exports.DeleteRepositoryCreationTemplateResponse$ = DeleteRepositoryCreationTemplateResponse$;
const DeleteRepositoryPolicyRequest$ = [3, n0, _DRPRel,
    0,
    [_rN, _rI],
    [0, 0], 1
];
exports.DeleteRepositoryPolicyRequest$ = DeleteRepositoryPolicyRequest$;
const DeleteRepositoryPolicyResponse$ = [3, n0, _DRPRele,
    0,
    [_rI, _rN, _pT],
    [0, 0, 0]
];
exports.DeleteRepositoryPolicyResponse$ = DeleteRepositoryPolicyResponse$;
const DeleteRepositoryRequest$ = [3, n0, _DRR,
    0,
    [_rN, _rI, _fo],
    [0, 0, 2], 1
];
exports.DeleteRepositoryRequest$ = DeleteRepositoryRequest$;
const DeleteRepositoryResponse$ = [3, n0, _DRRe,
    0,
    [_re],
    [() => Repository$]
];
exports.DeleteRepositoryResponse$ = DeleteRepositoryResponse$;
const DeleteSigningConfigurationRequest$ = [3, n0, _DSCR,
    0,
    [],
    []
];
exports.DeleteSigningConfigurationRequest$ = DeleteSigningConfigurationRequest$;
const DeleteSigningConfigurationResponse$ = [3, n0, _DSCRe,
    0,
    [_rI, _sCi],
    [0, () => SigningConfiguration$]
];
exports.DeleteSigningConfigurationResponse$ = DeleteSigningConfigurationResponse$;
const DeregisterPullTimeUpdateExclusionRequest$ = [3, n0, _DPTUER,
    0,
    [_pAr],
    [0], 1
];
exports.DeregisterPullTimeUpdateExclusionRequest$ = DeregisterPullTimeUpdateExclusionRequest$;
const DeregisterPullTimeUpdateExclusionResponse$ = [3, n0, _DPTUERe,
    0,
    [_pAr],
    [0]
];
exports.DeregisterPullTimeUpdateExclusionResponse$ = DeregisterPullTimeUpdateExclusionResponse$;
const DescribeImageReplicationStatusRequest$ = [3, n0, _DIRSR,
    0,
    [_rN, _iIm, _rI],
    [0, () => ImageIdentifier$, 0], 2
];
exports.DescribeImageReplicationStatusRequest$ = DescribeImageReplicationStatusRequest$;
const DescribeImageReplicationStatusResponse$ = [3, n0, _DIRSRe,
    0,
    [_rN, _iIm, _rS],
    [0, () => ImageIdentifier$, () => ImageReplicationStatusList]
];
exports.DescribeImageReplicationStatusResponse$ = DescribeImageReplicationStatusResponse$;
const DescribeImageScanFindingsRequest$ = [3, n0, _DISFR,
    0,
    [_rN, _iIm, _rI, _nT, _mR],
    [0, () => ImageIdentifier$, 0, 0, 1], 2
];
exports.DescribeImageScanFindingsRequest$ = DescribeImageScanFindingsRequest$;
const DescribeImageScanFindingsResponse$ = [3, n0, _DISFRe,
    0,
    [_rI, _rN, _iIm, _iSS, _iSF, _nT],
    [0, 0, () => ImageIdentifier$, () => ImageScanStatus$, () => ImageScanFindings$, 0]
];
exports.DescribeImageScanFindingsResponse$ = DescribeImageScanFindingsResponse$;
const DescribeImagesFilter$ = [3, n0, _DIF,
    0,
    [_tS, _iS],
    [0, 0]
];
exports.DescribeImagesFilter$ = DescribeImagesFilter$;
const DescribeImageSigningStatusRequest$ = [3, n0, _DISSR,
    0,
    [_rN, _iIm, _rI],
    [0, () => ImageIdentifier$, 0], 2
];
exports.DescribeImageSigningStatusRequest$ = DescribeImageSigningStatusRequest$;
const DescribeImageSigningStatusResponse$ = [3, n0, _DISSRe,
    0,
    [_rN, _iIm, _rI, _sSi],
    [0, () => ImageIdentifier$, 0, () => ImageSigningStatusList]
];
exports.DescribeImageSigningStatusResponse$ = DescribeImageSigningStatusResponse$;
const DescribeImagesRequest$ = [3, n0, _DIR,
    0,
    [_rN, _rI, _iI, _nT, _mR, _fi],
    [0, 0, () => ImageIdentifierList, 0, 1, () => DescribeImagesFilter$], 1
];
exports.DescribeImagesRequest$ = DescribeImagesRequest$;
const DescribeImagesResponse$ = [3, n0, _DIRe,
    0,
    [_iD, _nT],
    [() => ImageDetailList, 0]
];
exports.DescribeImagesResponse$ = DescribeImagesResponse$;
const DescribePullThroughCacheRulesRequest$ = [3, n0, _DPTCRRes,
    0,
    [_rI, _eRPc, _nT, _mR],
    [0, 64 | 0, 0, 1]
];
exports.DescribePullThroughCacheRulesRequest$ = DescribePullThroughCacheRulesRequest$;
const DescribePullThroughCacheRulesResponse$ = [3, n0, _DPTCRResc,
    0,
    [_pTCR, _nT],
    [() => PullThroughCacheRuleList, 0]
];
exports.DescribePullThroughCacheRulesResponse$ = DescribePullThroughCacheRulesResponse$;
const DescribeRegistryRequest$ = [3, n0, _DRRes,
    0,
    [],
    []
];
exports.DescribeRegistryRequest$ = DescribeRegistryRequest$;
const DescribeRegistryResponse$ = [3, n0, _DRResc,
    0,
    [_rI, _rC],
    [0, () => ReplicationConfiguration$]
];
exports.DescribeRegistryResponse$ = DescribeRegistryResponse$;
const DescribeRepositoriesRequest$ = [3, n0, _DRRescr,
    0,
    [_rI, _rNe, _nT, _mR],
    [0, 64 | 0, 0, 1]
];
exports.DescribeRepositoriesRequest$ = DescribeRepositoriesRequest$;
const DescribeRepositoriesResponse$ = [3, n0, _DRRescri,
    0,
    [_rep, _nT],
    [() => RepositoryList, 0]
];
exports.DescribeRepositoriesResponse$ = DescribeRepositoriesResponse$;
const DescribeRepositoryCreationTemplatesRequest$ = [3, n0, _DRCTRes,
    0,
    [_pre, _nT, _mR],
    [64 | 0, 0, 1]
];
exports.DescribeRepositoryCreationTemplatesRequest$ = DescribeRepositoryCreationTemplatesRequest$;
const DescribeRepositoryCreationTemplatesResponse$ = [3, n0, _DRCTResc,
    0,
    [_rI, _rCTe, _nT],
    [0, () => RepositoryCreationTemplateList, 0]
];
exports.DescribeRepositoryCreationTemplatesResponse$ = DescribeRepositoryCreationTemplatesResponse$;
const EncryptionConfiguration$ = [3, n0, _EC,
    0,
    [_eT, _kK],
    [0, 0], 1
];
exports.EncryptionConfiguration$ = EncryptionConfiguration$;
const EncryptionConfigurationForRepositoryCreationTemplate$ = [3, n0, _ECFRCT,
    0,
    [_eT, _kK],
    [0, 0], 1
];
exports.EncryptionConfigurationForRepositoryCreationTemplate$ = EncryptionConfigurationForRepositoryCreationTemplate$;
const EnhancedImageScanFinding$ = [3, n0, _EISF,
    0,
    [_aAI, _d, _fA, _fOA, _lOA, _pVD, _rem, _res, _sc, _sD, _sev, _st, _ti, _ty, _uA, _fAi, _eAx],
    [0, 0, 0, 4, 4, () => PackageVulnerabilityDetails$, () => Remediation$, () => ResourceList, 1, () => ScoreDetails$, 0, 0, 0, 0, 4, 0, 0]
];
exports.EnhancedImageScanFinding$ = EnhancedImageScanFinding$;
const GetAccountSettingRequest$ = [3, n0, _GASR,
    0,
    [_n],
    [0], 1
];
exports.GetAccountSettingRequest$ = GetAccountSettingRequest$;
const GetAccountSettingResponse$ = [3, n0, _GASRe,
    0,
    [_n, _v],
    [0, 0]
];
exports.GetAccountSettingResponse$ = GetAccountSettingResponse$;
const GetAuthorizationTokenRequest$ = [3, n0, _GATR,
    0,
    [_rIe],
    [64 | 0]
];
exports.GetAuthorizationTokenRequest$ = GetAuthorizationTokenRequest$;
const GetAuthorizationTokenResponse$ = [3, n0, _GATRe,
    0,
    [_aD],
    [() => AuthorizationDataList]
];
exports.GetAuthorizationTokenResponse$ = GetAuthorizationTokenResponse$;
const GetDownloadUrlForLayerRequest$ = [3, n0, _GDUFLR,
    0,
    [_rN, _lDa, _rI],
    [0, 0, 0], 2
];
exports.GetDownloadUrlForLayerRequest$ = GetDownloadUrlForLayerRequest$;
const GetDownloadUrlForLayerResponse$ = [3, n0, _GDUFLRe,
    0,
    [_dU, _lDa],
    [0, 0]
];
exports.GetDownloadUrlForLayerResponse$ = GetDownloadUrlForLayerResponse$;
const GetLifecyclePolicyPreviewRequest$ = [3, n0, _GLPPR,
    0,
    [_rN, _rI, _iI, _nT, _mR, _fi],
    [0, 0, () => ImageIdentifierList, 0, 1, () => LifecyclePolicyPreviewFilter$], 1
];
exports.GetLifecyclePolicyPreviewRequest$ = GetLifecyclePolicyPreviewRequest$;
const GetLifecyclePolicyPreviewResponse$ = [3, n0, _GLPPRe,
    0,
    [_rI, _rN, _lPT, _st, _nT, _pR, _su],
    [0, 0, 0, 0, 0, () => LifecyclePolicyPreviewResultList, () => LifecyclePolicyPreviewSummary$]
];
exports.GetLifecyclePolicyPreviewResponse$ = GetLifecyclePolicyPreviewResponse$;
const GetLifecyclePolicyRequest$ = [3, n0, _GLPR,
    0,
    [_rN, _rI],
    [0, 0], 1
];
exports.GetLifecyclePolicyRequest$ = GetLifecyclePolicyRequest$;
const GetLifecyclePolicyResponse$ = [3, n0, _GLPRe,
    0,
    [_rI, _rN, _lPT, _lEA],
    [0, 0, 0, 4]
];
exports.GetLifecyclePolicyResponse$ = GetLifecyclePolicyResponse$;
const GetRegistryPolicyRequest$ = [3, n0, _GRPR,
    0,
    [],
    []
];
exports.GetRegistryPolicyRequest$ = GetRegistryPolicyRequest$;
const GetRegistryPolicyResponse$ = [3, n0, _GRPRe,
    0,
    [_rI, _pT],
    [0, 0]
];
exports.GetRegistryPolicyResponse$ = GetRegistryPolicyResponse$;
const GetRegistryScanningConfigurationRequest$ = [3, n0, _GRSCR,
    0,
    [],
    []
];
exports.GetRegistryScanningConfigurationRequest$ = GetRegistryScanningConfigurationRequest$;
const GetRegistryScanningConfigurationResponse$ = [3, n0, _GRSCRe,
    0,
    [_rI, _sCc],
    [0, () => RegistryScanningConfiguration$]
];
exports.GetRegistryScanningConfigurationResponse$ = GetRegistryScanningConfigurationResponse$;
const GetRepositoryPolicyRequest$ = [3, n0, _GRPRet,
    0,
    [_rN, _rI],
    [0, 0], 1
];
exports.GetRepositoryPolicyRequest$ = GetRepositoryPolicyRequest$;
const GetRepositoryPolicyResponse$ = [3, n0, _GRPRete,
    0,
    [_rI, _rN, _pT],
    [0, 0, 0]
];
exports.GetRepositoryPolicyResponse$ = GetRepositoryPolicyResponse$;
const GetSigningConfigurationRequest$ = [3, n0, _GSCR,
    0,
    [],
    []
];
exports.GetSigningConfigurationRequest$ = GetSigningConfigurationRequest$;
const GetSigningConfigurationResponse$ = [3, n0, _GSCRe,
    0,
    [_rI, _sCi],
    [0, () => SigningConfiguration$]
];
exports.GetSigningConfigurationResponse$ = GetSigningConfigurationResponse$;
const Image$ = [3, n0, _I,
    0,
    [_rI, _rN, _iIm, _iM, _iMMT],
    [0, 0, () => ImageIdentifier$, 0, 0]
];
exports.Image$ = Image$;
const ImageDetail$ = [3, n0, _ID,
    0,
    [_rI, _rN, _iDm, _iT, _iSIB, _iPA, _iSS, _iSFS, _iMMT, _aMTr, _lRPT, _sMD, _iS, _lAA, _lAAa],
    [0, 0, 0, 64 | 0, 1, 4, () => ImageScanStatus$, () => ImageScanFindingsSummary$, 0, 0, 4, 0, 0, 4, 4]
];
exports.ImageDetail$ = ImageDetail$;
const ImageFailure$ = [3, n0, _IF,
    0,
    [_iIm, _fC, _fR],
    [() => ImageIdentifier$, 0, 0]
];
exports.ImageFailure$ = ImageFailure$;
const ImageIdentifier$ = [3, n0, _II,
    0,
    [_iDm, _iTm],
    [0, 0]
];
exports.ImageIdentifier$ = ImageIdentifier$;
const ImageReferrer$ = [3, n0, _IR,
    0,
    [_di, _mT, _si, _aTr, _an, _aS],
    [0, 0, 1, 0, 128 | 0, 0], 3
];
exports.ImageReferrer$ = ImageReferrer$;
const ImageReplicationStatus$ = [3, n0, _IRS,
    0,
    [_reg, _rI, _st, _fC],
    [0, 0, 0, 0]
];
exports.ImageReplicationStatus$ = ImageReplicationStatus$;
const ImageScanFinding$ = [3, n0, _ISF,
    0,
    [_n, _d, _u, _sev, _at],
    [0, 0, 0, 0, () => AttributeList]
];
exports.ImageScanFinding$ = ImageScanFinding$;
const ImageScanFindings$ = [3, n0, _ISFm,
    0,
    [_iSCA, _vSUA, _fSC, _fin, _eF],
    [4, 4, 128 | 1, () => ImageScanFindingList, () => EnhancedImageScanFindingList]
];
exports.ImageScanFindings$ = ImageScanFindings$;
const ImageScanFindingsSummary$ = [3, n0, _ISFS,
    0,
    [_iSCA, _vSUA, _fSC],
    [4, 4, 128 | 1]
];
exports.ImageScanFindingsSummary$ = ImageScanFindingsSummary$;
const ImageScanningConfiguration$ = [3, n0, _ISC,
    0,
    [_sOP],
    [2]
];
exports.ImageScanningConfiguration$ = ImageScanningConfiguration$;
const ImageScanStatus$ = [3, n0, _ISS,
    0,
    [_st, _d],
    [0, 0]
];
exports.ImageScanStatus$ = ImageScanStatus$;
const ImageSigningStatus$ = [3, n0, _ISSm,
    0,
    [_sPA, _fC, _fR, _st],
    [0, 0, 0, 0]
];
exports.ImageSigningStatus$ = ImageSigningStatus$;
const ImageTagMutabilityExclusionFilter$ = [3, n0, _ITMEF,
    0,
    [_fT, _fi],
    [0, 0], 2
];
exports.ImageTagMutabilityExclusionFilter$ = ImageTagMutabilityExclusionFilter$;
const InitiateLayerUploadRequest$ = [3, n0, _ILUR,
    0,
    [_rN, _rI],
    [0, 0], 1
];
exports.InitiateLayerUploadRequest$ = InitiateLayerUploadRequest$;
const InitiateLayerUploadResponse$ = [3, n0, _ILURn,
    0,
    [_uI, _pS],
    [0, 1]
];
exports.InitiateLayerUploadResponse$ = InitiateLayerUploadResponse$;
const Layer$ = [3, n0, _L,
    0,
    [_lDa, _lA, _lS, _mT],
    [0, 0, 1, 0]
];
exports.Layer$ = Layer$;
const LayerFailure$ = [3, n0, _LF,
    0,
    [_lDa, _fC, _fR],
    [0, 0, 0]
];
exports.LayerFailure$ = LayerFailure$;
const LifecyclePolicyPreviewFilter$ = [3, n0, _LPPF,
    0,
    [_tS],
    [0]
];
exports.LifecyclePolicyPreviewFilter$ = LifecyclePolicyPreviewFilter$;
const LifecyclePolicyPreviewResult$ = [3, n0, _LPPR,
    0,
    [_iT, _iDm, _iPA, _ac, _aRP, _sCt],
    [64 | 0, 0, 4, () => LifecyclePolicyRuleAction$, 1, 0]
];
exports.LifecyclePolicyPreviewResult$ = LifecyclePolicyPreviewResult$;
const LifecyclePolicyPreviewSummary$ = [3, n0, _LPPS,
    0,
    [_eITC, _tITC],
    [1, () => TransitioningImageTotalCounts]
];
exports.LifecyclePolicyPreviewSummary$ = LifecyclePolicyPreviewSummary$;
const LifecyclePolicyRuleAction$ = [3, n0, _LPRA,
    0,
    [_ty, _tSC],
    [0, 0]
];
exports.LifecyclePolicyRuleAction$ = LifecyclePolicyRuleAction$;
const ListImageReferrersFilter$ = [3, n0, _LIRF,
    0,
    [_aTrt, _aS],
    [64 | 0, 0]
];
exports.ListImageReferrersFilter$ = ListImageReferrersFilter$;
const ListImageReferrersRequest$ = [3, n0, _LIRR,
    0,
    [_rN, _sI, _rI, _fi, _nT, _mR],
    [0, () => SubjectIdentifier$, 0, () => ListImageReferrersFilter$, 0, 1], 2
];
exports.ListImageReferrersRequest$ = ListImageReferrersRequest$;
const ListImageReferrersResponse$ = [3, n0, _LIRRi,
    0,
    [_ref, _nT],
    [() => ImageReferrerList, 0]
];
exports.ListImageReferrersResponse$ = ListImageReferrersResponse$;
const ListImagesFilter$ = [3, n0, _LIF,
    0,
    [_tS, _iS],
    [0, 0]
];
exports.ListImagesFilter$ = ListImagesFilter$;
const ListImagesRequest$ = [3, n0, _LIR,
    0,
    [_rN, _rI, _nT, _mR, _fi],
    [0, 0, 0, 1, () => ListImagesFilter$], 1
];
exports.ListImagesRequest$ = ListImagesRequest$;
const ListImagesResponse$ = [3, n0, _LIRi,
    0,
    [_iI, _nT],
    [() => ImageIdentifierList, 0]
];
exports.ListImagesResponse$ = ListImagesResponse$;
const ListPullTimeUpdateExclusionsRequest$ = [3, n0, _LPTUER,
    0,
    [_mR, _nT],
    [1, 0]
];
exports.ListPullTimeUpdateExclusionsRequest$ = ListPullTimeUpdateExclusionsRequest$;
const ListPullTimeUpdateExclusionsResponse$ = [3, n0, _LPTUERi,
    0,
    [_pTUE, _nT],
    [64 | 0, 0]
];
exports.ListPullTimeUpdateExclusionsResponse$ = ListPullTimeUpdateExclusionsResponse$;
const ListTagsForResourceRequest$ = [3, n0, _LTFRR,
    0,
    [_rA],
    [0], 1
];
exports.ListTagsForResourceRequest$ = ListTagsForResourceRequest$;
const ListTagsForResourceResponse$ = [3, n0, _LTFRRi,
    0,
    [_t],
    [() => TagList]
];
exports.ListTagsForResourceResponse$ = ListTagsForResourceResponse$;
const PackageVulnerabilityDetails$ = [3, n0, _PVD,
    0,
    [_cv, _rU, _rV, _so, _sU, _vCA, _vS, _vUA, _vI, _vP],
    [() => CvssScoreList, 64 | 0, 64 | 0, 0, 0, 4, 0, 4, 0, () => VulnerablePackagesList]
];
exports.PackageVulnerabilityDetails$ = PackageVulnerabilityDetails$;
const PullThroughCacheRule$ = [3, n0, _PTCR,
    0,
    [_eRP, _uRU, _cAr, _rI, _cA, _cRA, _uRP, _uR, _uA],
    [0, 0, 4, 0, 0, 0, 0, 0, 4]
];
exports.PullThroughCacheRule$ = PullThroughCacheRule$;
const PutAccountSettingRequest$ = [3, n0, _PASR,
    0,
    [_n, _v],
    [0, 0], 2
];
exports.PutAccountSettingRequest$ = PutAccountSettingRequest$;
const PutAccountSettingResponse$ = [3, n0, _PASRu,
    0,
    [_n, _v],
    [0, 0]
];
exports.PutAccountSettingResponse$ = PutAccountSettingResponse$;
const PutImageRequest$ = [3, n0, _PIR,
    0,
    [_rN, _iM, _rI, _iMMT, _iTm, _iDm],
    [0, 0, 0, 0, 0, 0], 2
];
exports.PutImageRequest$ = PutImageRequest$;
const PutImageResponse$ = [3, n0, _PIRu,
    0,
    [_im],
    [() => Image$]
];
exports.PutImageResponse$ = PutImageResponse$;
const PutImageScanningConfigurationRequest$ = [3, n0, _PISCR,
    0,
    [_rN, _iSC, _rI],
    [0, () => ImageScanningConfiguration$, 0], 2
];
exports.PutImageScanningConfigurationRequest$ = PutImageScanningConfigurationRequest$;
const PutImageScanningConfigurationResponse$ = [3, n0, _PISCRu,
    0,
    [_rI, _rN, _iSC],
    [0, 0, () => ImageScanningConfiguration$]
];
exports.PutImageScanningConfigurationResponse$ = PutImageScanningConfigurationResponse$;
const PutImageTagMutabilityRequest$ = [3, n0, _PITMR,
    0,
    [_rN, _iTM, _rI, _iTMEF],
    [0, 0, 0, () => ImageTagMutabilityExclusionFilters], 2
];
exports.PutImageTagMutabilityRequest$ = PutImageTagMutabilityRequest$;
const PutImageTagMutabilityResponse$ = [3, n0, _PITMRu,
    0,
    [_rI, _rN, _iTM, _iTMEF],
    [0, 0, 0, () => ImageTagMutabilityExclusionFilters]
];
exports.PutImageTagMutabilityResponse$ = PutImageTagMutabilityResponse$;
const PutLifecyclePolicyRequest$ = [3, n0, _PLPR,
    0,
    [_rN, _lPT, _rI],
    [0, 0, 0], 2
];
exports.PutLifecyclePolicyRequest$ = PutLifecyclePolicyRequest$;
const PutLifecyclePolicyResponse$ = [3, n0, _PLPRu,
    0,
    [_rI, _rN, _lPT],
    [0, 0, 0]
];
exports.PutLifecyclePolicyResponse$ = PutLifecyclePolicyResponse$;
const PutRegistryPolicyRequest$ = [3, n0, _PRPR,
    0,
    [_pT],
    [0], 1
];
exports.PutRegistryPolicyRequest$ = PutRegistryPolicyRequest$;
const PutRegistryPolicyResponse$ = [3, n0, _PRPRu,
    0,
    [_rI, _pT],
    [0, 0]
];
exports.PutRegistryPolicyResponse$ = PutRegistryPolicyResponse$;
const PutRegistryScanningConfigurationRequest$ = [3, n0, _PRSCR,
    0,
    [_sT, _ru],
    [0, () => RegistryScanningRuleList]
];
exports.PutRegistryScanningConfigurationRequest$ = PutRegistryScanningConfigurationRequest$;
const PutRegistryScanningConfigurationResponse$ = [3, n0, _PRSCRu,
    0,
    [_rSC],
    [() => RegistryScanningConfiguration$]
];
exports.PutRegistryScanningConfigurationResponse$ = PutRegistryScanningConfigurationResponse$;
const PutReplicationConfigurationRequest$ = [3, n0, _PRCR,
    0,
    [_rC],
    [() => ReplicationConfiguration$], 1
];
exports.PutReplicationConfigurationRequest$ = PutReplicationConfigurationRequest$;
const PutReplicationConfigurationResponse$ = [3, n0, _PRCRu,
    0,
    [_rC],
    [() => ReplicationConfiguration$]
];
exports.PutReplicationConfigurationResponse$ = PutReplicationConfigurationResponse$;
const PutSigningConfigurationRequest$ = [3, n0, _PSCR,
    0,
    [_sCi],
    [() => SigningConfiguration$], 1
];
exports.PutSigningConfigurationRequest$ = PutSigningConfigurationRequest$;
const PutSigningConfigurationResponse$ = [3, n0, _PSCRu,
    0,
    [_sCi],
    [() => SigningConfiguration$]
];
exports.PutSigningConfigurationResponse$ = PutSigningConfigurationResponse$;
const Recommendation$ = [3, n0, _R,
    0,
    [_ur, _te],
    [0, 0]
];
exports.Recommendation$ = Recommendation$;
const RegisterPullTimeUpdateExclusionRequest$ = [3, n0, _RPTUER,
    0,
    [_pAr],
    [0], 1
];
exports.RegisterPullTimeUpdateExclusionRequest$ = RegisterPullTimeUpdateExclusionRequest$;
const RegisterPullTimeUpdateExclusionResponse$ = [3, n0, _RPTUERe,
    0,
    [_pAr, _cAr],
    [0, 4]
];
exports.RegisterPullTimeUpdateExclusionResponse$ = RegisterPullTimeUpdateExclusionResponse$;
const RegistryScanningConfiguration$ = [3, n0, _RSC,
    0,
    [_sT, _ru],
    [0, () => RegistryScanningRuleList]
];
exports.RegistryScanningConfiguration$ = RegistryScanningConfiguration$;
const RegistryScanningRule$ = [3, n0, _RSR,
    0,
    [_sF, _rF],
    [0, () => ScanningRepositoryFilterList], 2
];
exports.RegistryScanningRule$ = RegistryScanningRule$;
const Remediation$ = [3, n0, _Re,
    0,
    [_rec],
    [() => Recommendation$]
];
exports.Remediation$ = Remediation$;
const ReplicationConfiguration$ = [3, n0, _RC,
    0,
    [_ru],
    [() => ReplicationRuleList], 1
];
exports.ReplicationConfiguration$ = ReplicationConfiguration$;
const ReplicationDestination$ = [3, n0, _RD,
    0,
    [_reg, _rI],
    [0, 0], 2
];
exports.ReplicationDestination$ = ReplicationDestination$;
const ReplicationRule$ = [3, n0, _RR,
    0,
    [_de, _rF],
    [() => ReplicationDestinationList, () => RepositoryFilterList], 1
];
exports.ReplicationRule$ = ReplicationRule$;
const Repository$ = [3, n0, _Rep,
    0,
    [_rAe, _rI, _rN, _rUe, _cAr, _iTM, _iTMEF, _iSC, _eC],
    [0, 0, 0, 0, 4, 0, () => ImageTagMutabilityExclusionFilters, () => ImageScanningConfiguration$, () => EncryptionConfiguration$]
];
exports.Repository$ = Repository$;
const RepositoryCreationTemplate$ = [3, n0, _RCT,
    0,
    [_pr, _d, _eC, _rT, _iTM, _iTMEF, _rP, _lP, _aF, _cRA, _cAr, _uA],
    [0, 0, () => EncryptionConfigurationForRepositoryCreationTemplate$, () => TagList, 0, () => ImageTagMutabilityExclusionFilters, 0, 0, 64 | 0, 0, 4, 4]
];
exports.RepositoryCreationTemplate$ = RepositoryCreationTemplate$;
const RepositoryFilter$ = [3, n0, _RF,
    0,
    [_fi, _fT],
    [0, 0], 2
];
exports.RepositoryFilter$ = RepositoryFilter$;
const RepositoryScanningConfiguration$ = [3, n0, _RSCe,
    0,
    [_rAe, _rN, _sOP, _sF, _aSF],
    [0, 0, 2, 0, () => ScanningRepositoryFilterList]
];
exports.RepositoryScanningConfiguration$ = RepositoryScanningConfiguration$;
const RepositoryScanningConfigurationFailure$ = [3, n0, _RSCF,
    0,
    [_rN, _fC, _fR],
    [0, 0, 0]
];
exports.RepositoryScanningConfigurationFailure$ = RepositoryScanningConfigurationFailure$;
const Resource$ = [3, n0, _Res,
    0,
    [_det, _id, _t, _ty],
    [() => ResourceDetails$, 0, 128 | 0, 0]
];
exports.Resource$ = Resource$;
const ResourceDetails$ = [3, n0, _RDe,
    0,
    [_aECI],
    [() => AwsEcrContainerImageDetails$]
];
exports.ResourceDetails$ = ResourceDetails$;
const ScanningRepositoryFilter$ = [3, n0, _SRF,
    0,
    [_fi, _fT],
    [0, 0], 2
];
exports.ScanningRepositoryFilter$ = ScanningRepositoryFilter$;
const ScoreDetails$ = [3, n0, _SD,
    0,
    [_cv],
    [() => CvssScoreDetails$]
];
exports.ScoreDetails$ = ScoreDetails$;
const SetRepositoryPolicyRequest$ = [3, n0, _SRPR,
    0,
    [_rN, _pT, _rI, _fo],
    [0, 0, 0, 2], 2
];
exports.SetRepositoryPolicyRequest$ = SetRepositoryPolicyRequest$;
const SetRepositoryPolicyResponse$ = [3, n0, _SRPRe,
    0,
    [_rI, _rN, _pT],
    [0, 0, 0]
];
exports.SetRepositoryPolicyResponse$ = SetRepositoryPolicyResponse$;
const SigningConfiguration$ = [3, n0, _SC,
    0,
    [_ru],
    [() => SigningRuleList], 1
];
exports.SigningConfiguration$ = SigningConfiguration$;
const SigningRepositoryFilter$ = [3, n0, _SRFi,
    0,
    [_fi, _fT],
    [0, 0], 2
];
exports.SigningRepositoryFilter$ = SigningRepositoryFilter$;
const SigningRule$ = [3, n0, _SR,
    0,
    [_sPA, _rF],
    [0, () => SigningRepositoryFilterList], 1
];
exports.SigningRule$ = SigningRule$;
const StartImageScanRequest$ = [3, n0, _SISR,
    0,
    [_rN, _iIm, _rI],
    [0, () => ImageIdentifier$, 0], 2
];
exports.StartImageScanRequest$ = StartImageScanRequest$;
const StartImageScanResponse$ = [3, n0, _SISRt,
    0,
    [_rI, _rN, _iIm, _iSS],
    [0, 0, () => ImageIdentifier$, () => ImageScanStatus$]
];
exports.StartImageScanResponse$ = StartImageScanResponse$;
const StartLifecyclePolicyPreviewRequest$ = [3, n0, _SLPPR,
    0,
    [_rN, _rI, _lPT],
    [0, 0, 0], 1
];
exports.StartLifecyclePolicyPreviewRequest$ = StartLifecyclePolicyPreviewRequest$;
const StartLifecyclePolicyPreviewResponse$ = [3, n0, _SLPPRt,
    0,
    [_rI, _rN, _lPT, _st],
    [0, 0, 0, 0]
];
exports.StartLifecyclePolicyPreviewResponse$ = StartLifecyclePolicyPreviewResponse$;
const SubjectIdentifier$ = [3, n0, _SI,
    0,
    [_iDm],
    [0], 1
];
exports.SubjectIdentifier$ = SubjectIdentifier$;
const Tag$ = [3, n0, _T,
    0,
    [_K, _V],
    [0, 0], 2
];
exports.Tag$ = Tag$;
const TagResourceRequest$ = [3, n0, _TRR,
    0,
    [_rA, _t],
    [0, () => TagList], 2
];
exports.TagResourceRequest$ = TagResourceRequest$;
const TagResourceResponse$ = [3, n0, _TRRa,
    0,
    [],
    []
];
exports.TagResourceResponse$ = TagResourceResponse$;
const TransitioningImageTotalCount$ = [3, n0, _TITC,
    0,
    [_tSC, _iTC],
    [0, 1]
];
exports.TransitioningImageTotalCount$ = TransitioningImageTotalCount$;
const UntagResourceRequest$ = [3, n0, _URR,
    0,
    [_rA, _tK],
    [0, 64 | 0], 2
];
exports.UntagResourceRequest$ = UntagResourceRequest$;
const UntagResourceResponse$ = [3, n0, _URRn,
    0,
    [],
    []
];
exports.UntagResourceResponse$ = UntagResourceResponse$;
const UpdateImageStorageClassRequest$ = [3, n0, _UISCR,
    0,
    [_rN, _iIm, _tSC, _rI],
    [0, () => ImageIdentifier$, 0, 0], 3
];
exports.UpdateImageStorageClassRequest$ = UpdateImageStorageClassRequest$;
const UpdateImageStorageClassResponse$ = [3, n0, _UISCRp,
    0,
    [_rI, _rN, _iIm, _iS],
    [0, 0, () => ImageIdentifier$, 0]
];
exports.UpdateImageStorageClassResponse$ = UpdateImageStorageClassResponse$;
const UpdatePullThroughCacheRuleRequest$ = [3, n0, _UPTCRR,
    0,
    [_eRP, _rI, _cA, _cRA],
    [0, 0, 0, 0], 1
];
exports.UpdatePullThroughCacheRuleRequest$ = UpdatePullThroughCacheRuleRequest$;
const UpdatePullThroughCacheRuleResponse$ = [3, n0, _UPTCRRp,
    0,
    [_eRP, _rI, _uA, _cA, _cRA, _uRP],
    [0, 0, 4, 0, 0, 0]
];
exports.UpdatePullThroughCacheRuleResponse$ = UpdatePullThroughCacheRuleResponse$;
const UpdateRepositoryCreationTemplateRequest$ = [3, n0, _URCTR,
    0,
    [_pr, _d, _eC, _rT, _iTM, _iTMEF, _rP, _lP, _aF, _cRA],
    [0, 0, () => EncryptionConfigurationForRepositoryCreationTemplate$, () => TagList, 0, () => ImageTagMutabilityExclusionFilters, 0, 0, 64 | 0, 0], 1
];
exports.UpdateRepositoryCreationTemplateRequest$ = UpdateRepositoryCreationTemplateRequest$;
const UpdateRepositoryCreationTemplateResponse$ = [3, n0, _URCTRp,
    0,
    [_rI, _rCT],
    [0, () => RepositoryCreationTemplate$]
];
exports.UpdateRepositoryCreationTemplateResponse$ = UpdateRepositoryCreationTemplateResponse$;
const UploadLayerPartRequest$ = [3, n0, _ULPR,
    0,
    [_rN, _uI, _pFB, _pLB, _lPB, _rI],
    [0, 0, 1, 1, 21, 0], 5
];
exports.UploadLayerPartRequest$ = UploadLayerPartRequest$;
const UploadLayerPartResponse$ = [3, n0, _ULPRp,
    0,
    [_rI, _rN, _uI, _lBR],
    [0, 0, 0, 1]
];
exports.UploadLayerPartResponse$ = UploadLayerPartResponse$;
const ValidatePullThroughCacheRuleRequest$ = [3, n0, _VPTCRR,
    0,
    [_eRP, _rI],
    [0, 0], 1
];
exports.ValidatePullThroughCacheRuleRequest$ = ValidatePullThroughCacheRuleRequest$;
const ValidatePullThroughCacheRuleResponse$ = [3, n0, _VPTCRRa,
    0,
    [_eRP, _rI, _uRU, _cA, _cRA, _uRP, _iV, _fa],
    [0, 0, 0, 0, 0, 0, 2, 0]
];
exports.ValidatePullThroughCacheRuleResponse$ = ValidatePullThroughCacheRuleResponse$;
const VulnerablePackage$ = [3, n0, _VP,
    0,
    [_ar, _ep, _fP, _n, _pM, _rel, _sLH, _ve, _fIV],
    [0, 1, 0, 0, 0, 0, 0, 0, 0]
];
exports.VulnerablePackage$ = VulnerablePackage$;
var ArtifactTypeList = 64 | 0;
var AttributeList = [1, n0, _AL,
    0, () => Attribute$
];
var AuthorizationDataList = [1, n0, _ADL,
    0, () => AuthorizationData$
];
var BatchedOperationLayerDigestList = 64 | 0;
var CvssScoreAdjustmentList = [1, n0, _CSAL,
    0, () => CvssScoreAdjustment$
];
var CvssScoreList = [1, n0, _CSL,
    0, () => CvssScore$
];
var EnhancedImageScanFindingList = [1, n0, _EISFL,
    0, () => EnhancedImageScanFinding$
];
var GetAuthorizationTokenRegistryIdList = 64 | 0;
var ImageDetailList = [1, n0, _IDL,
    0, () => ImageDetail$
];
var ImageFailureList = [1, n0, _IFL,
    0, () => ImageFailure$
];
var ImageIdentifierList = [1, n0, _IIL,
    0, () => ImageIdentifier$
];
var ImageList = [1, n0, _IL,
    0, () => Image$
];
var ImageReferrerList = [1, n0, _IRL,
    0, () => ImageReferrer$
];
var ImageReplicationStatusList = [1, n0, _IRSL,
    0, () => ImageReplicationStatus$
];
var ImageScanFindingList = [1, n0, _ISFL,
    0, () => ImageScanFinding$
];
var ImageSigningStatusList = [1, n0, _ISSL,
    0, () => ImageSigningStatus$
];
var ImageTagList = 64 | 0;
var ImageTagMutabilityExclusionFilters = [1, n0, _ITMEFm,
    0, () => ImageTagMutabilityExclusionFilter$
];
var ImageTagsList = 64 | 0;
var LayerDigestList = 64 | 0;
var LayerFailureList = [1, n0, _LFL,
    0, () => LayerFailure$
];
var LayerList = [1, n0, _LL,
    0, () => Layer$
];
var LifecyclePolicyPreviewResultList = [1, n0, _LPPRL,
    0, () => LifecyclePolicyPreviewResult$
];
var MediaTypeList = 64 | 0;
var PrefixList = 64 | 0;
var PullThroughCacheRuleList = [1, n0, _PTCRL,
    0, () => PullThroughCacheRule$
];
var PullThroughCacheRuleRepositoryPrefixList = 64 | 0;
var PullTimeUpdateExclusionList = 64 | 0;
var RCTAppliedForList = 64 | 0;
var ReferenceUrlsList = 64 | 0;
var RegistryScanningRuleList = [1, n0, _RSRL,
    0, () => RegistryScanningRule$
];
var RelatedVulnerabilitiesList = 64 | 0;
var ReplicationDestinationList = [1, n0, _RDL,
    0, () => ReplicationDestination$
];
var ReplicationRuleList = [1, n0, _RRL,
    0, () => ReplicationRule$
];
var RepositoryCreationTemplateList = [1, n0, _RCTL,
    0, () => RepositoryCreationTemplate$
];
var RepositoryFilterList = [1, n0, _RFL,
    0, () => RepositoryFilter$
];
var RepositoryList = [1, n0, _RL,
    0, () => Repository$
];
var RepositoryNameList = 64 | 0;
var RepositoryScanningConfigurationFailureList = [1, n0, _RSCFL,
    0, () => RepositoryScanningConfigurationFailure$
];
var RepositoryScanningConfigurationList = [1, n0, _RSCL,
    0, () => RepositoryScanningConfiguration$
];
var ResourceList = [1, n0, _RLe,
    0, () => Resource$
];
var ScanningConfigurationRepositoryNameList = 64 | 0;
var ScanningRepositoryFilterList = [1, n0, _SRFL,
    0, () => ScanningRepositoryFilter$
];
var SigningRepositoryFilterList = [1, n0, _SRFLi,
    0, () => SigningRepositoryFilter$
];
var SigningRuleList = [1, n0, _SRL,
    0, () => SigningRule$
];
var TagKeyList = 64 | 0;
var TagList = [1, n0, _TL,
    0, () => Tag$
];
var TransitioningImageTotalCounts = [1, n0, _TITCr,
    0, () => TransitioningImageTotalCount$
];
var VulnerablePackagesList = [1, n0, _VPL,
    0, () => VulnerablePackage$
];
var Annotations = 128 | 0;
var FindingSeverityCounts = 128 | 1;
var Tags = 128 | 0;
exports.BatchCheckLayerAvailability$ = [9, n0, _BCLA,
    0, () => BatchCheckLayerAvailabilityRequest$, () => BatchCheckLayerAvailabilityResponse$
];
exports.BatchDeleteImage$ = [9, n0, _BDI,
    0, () => BatchDeleteImageRequest$, () => BatchDeleteImageResponse$
];
exports.BatchGetImage$ = [9, n0, _BGI,
    0, () => BatchGetImageRequest$, () => BatchGetImageResponse$
];
exports.BatchGetRepositoryScanningConfiguration$ = [9, n0, _BGRSC,
    0, () => BatchGetRepositoryScanningConfigurationRequest$, () => BatchGetRepositoryScanningConfigurationResponse$
];
exports.CompleteLayerUpload$ = [9, n0, _CLU,
    0, () => CompleteLayerUploadRequest$, () => CompleteLayerUploadResponse$
];
exports.CreatePullThroughCacheRule$ = [9, n0, _CPTCR,
    0, () => CreatePullThroughCacheRuleRequest$, () => CreatePullThroughCacheRuleResponse$
];
exports.CreateRepository$ = [9, n0, _CR,
    0, () => CreateRepositoryRequest$, () => CreateRepositoryResponse$
];
exports.CreateRepositoryCreationTemplate$ = [9, n0, _CRCT,
    0, () => CreateRepositoryCreationTemplateRequest$, () => CreateRepositoryCreationTemplateResponse$
];
exports.DeleteLifecyclePolicy$ = [9, n0, _DLP,
    0, () => DeleteLifecyclePolicyRequest$, () => DeleteLifecyclePolicyResponse$
];
exports.DeletePullThroughCacheRule$ = [9, n0, _DPTCR,
    0, () => DeletePullThroughCacheRuleRequest$, () => DeletePullThroughCacheRuleResponse$
];
exports.DeleteRegistryPolicy$ = [9, n0, _DRP,
    0, () => DeleteRegistryPolicyRequest$, () => DeleteRegistryPolicyResponse$
];
exports.DeleteRepository$ = [9, n0, _DR,
    0, () => DeleteRepositoryRequest$, () => DeleteRepositoryResponse$
];
exports.DeleteRepositoryCreationTemplate$ = [9, n0, _DRCT,
    0, () => DeleteRepositoryCreationTemplateRequest$, () => DeleteRepositoryCreationTemplateResponse$
];
exports.DeleteRepositoryPolicy$ = [9, n0, _DRPe,
    0, () => DeleteRepositoryPolicyRequest$, () => DeleteRepositoryPolicyResponse$
];
exports.DeleteSigningConfiguration$ = [9, n0, _DSC,
    0, () => DeleteSigningConfigurationRequest$, () => DeleteSigningConfigurationResponse$
];
exports.DeregisterPullTimeUpdateExclusion$ = [9, n0, _DPTUE,
    0, () => DeregisterPullTimeUpdateExclusionRequest$, () => DeregisterPullTimeUpdateExclusionResponse$
];
exports.DescribeImageReplicationStatus$ = [9, n0, _DIRS,
    0, () => DescribeImageReplicationStatusRequest$, () => DescribeImageReplicationStatusResponse$
];
exports.DescribeImages$ = [9, n0, _DI,
    0, () => DescribeImagesRequest$, () => DescribeImagesResponse$
];
exports.DescribeImageScanFindings$ = [9, n0, _DISF,
    0, () => DescribeImageScanFindingsRequest$, () => DescribeImageScanFindingsResponse$
];
exports.DescribeImageSigningStatus$ = [9, n0, _DISS,
    0, () => DescribeImageSigningStatusRequest$, () => DescribeImageSigningStatusResponse$
];
exports.DescribePullThroughCacheRules$ = [9, n0, _DPTCRe,
    0, () => DescribePullThroughCacheRulesRequest$, () => DescribePullThroughCacheRulesResponse$
];
exports.DescribeRegistry$ = [9, n0, _DRe,
    0, () => DescribeRegistryRequest$, () => DescribeRegistryResponse$
];
exports.DescribeRepositories$ = [9, n0, _DRes,
    0, () => DescribeRepositoriesRequest$, () => DescribeRepositoriesResponse$
];
exports.DescribeRepositoryCreationTemplates$ = [9, n0, _DRCTe,
    0, () => DescribeRepositoryCreationTemplatesRequest$, () => DescribeRepositoryCreationTemplatesResponse$
];
exports.GetAccountSetting$ = [9, n0, _GAS,
    0, () => GetAccountSettingRequest$, () => GetAccountSettingResponse$
];
exports.GetAuthorizationToken$ = [9, n0, _GAT,
    0, () => GetAuthorizationTokenRequest$, () => GetAuthorizationTokenResponse$
];
exports.GetDownloadUrlForLayer$ = [9, n0, _GDUFL,
    0, () => GetDownloadUrlForLayerRequest$, () => GetDownloadUrlForLayerResponse$
];
exports.GetLifecyclePolicy$ = [9, n0, _GLP,
    0, () => GetLifecyclePolicyRequest$, () => GetLifecyclePolicyResponse$
];
exports.GetLifecyclePolicyPreview$ = [9, n0, _GLPP,
    0, () => GetLifecyclePolicyPreviewRequest$, () => GetLifecyclePolicyPreviewResponse$
];
exports.GetRegistryPolicy$ = [9, n0, _GRP,
    0, () => GetRegistryPolicyRequest$, () => GetRegistryPolicyResponse$
];
exports.GetRegistryScanningConfiguration$ = [9, n0, _GRSC,
    0, () => GetRegistryScanningConfigurationRequest$, () => GetRegistryScanningConfigurationResponse$
];
exports.GetRepositoryPolicy$ = [9, n0, _GRPe,
    0, () => GetRepositoryPolicyRequest$, () => GetRepositoryPolicyResponse$
];
exports.GetSigningConfiguration$ = [9, n0, _GSC,
    0, () => GetSigningConfigurationRequest$, () => GetSigningConfigurationResponse$
];
exports.InitiateLayerUpload$ = [9, n0, _ILU,
    0, () => InitiateLayerUploadRequest$, () => InitiateLayerUploadResponse$
];
exports.ListImageReferrers$ = [9, n0, _LIRis,
    0, () => ListImageReferrersRequest$, () => ListImageReferrersResponse$
];
exports.ListImages$ = [9, n0, _LI,
    0, () => ListImagesRequest$, () => ListImagesResponse$
];
exports.ListPullTimeUpdateExclusions$ = [9, n0, _LPTUE,
    0, () => ListPullTimeUpdateExclusionsRequest$, () => ListPullTimeUpdateExclusionsResponse$
];
exports.ListTagsForResource$ = [9, n0, _LTFR,
    0, () => ListTagsForResourceRequest$, () => ListTagsForResourceResponse$
];
exports.PutAccountSetting$ = [9, n0, _PAS,
    0, () => PutAccountSettingRequest$, () => PutAccountSettingResponse$
];
exports.PutImage$ = [9, n0, _PI,
    0, () => PutImageRequest$, () => PutImageResponse$
];
exports.PutImageScanningConfiguration$ = [9, n0, _PISC,
    0, () => PutImageScanningConfigurationRequest$, () => PutImageScanningConfigurationResponse$
];
exports.PutImageTagMutability$ = [9, n0, _PITM,
    0, () => PutImageTagMutabilityRequest$, () => PutImageTagMutabilityResponse$
];
exports.PutLifecyclePolicy$ = [9, n0, _PLP,
    0, () => PutLifecyclePolicyRequest$, () => PutLifecyclePolicyResponse$
];
exports.PutRegistryPolicy$ = [9, n0, _PRP,
    0, () => PutRegistryPolicyRequest$, () => PutRegistryPolicyResponse$
];
exports.PutRegistryScanningConfiguration$ = [9, n0, _PRSC,
    0, () => PutRegistryScanningConfigurationRequest$, () => PutRegistryScanningConfigurationResponse$
];
exports.PutReplicationConfiguration$ = [9, n0, _PRC,
    0, () => PutReplicationConfigurationRequest$, () => PutReplicationConfigurationResponse$
];
exports.PutSigningConfiguration$ = [9, n0, _PSC,
    0, () => PutSigningConfigurationRequest$, () => PutSigningConfigurationResponse$
];
exports.RegisterPullTimeUpdateExclusion$ = [9, n0, _RPTUE,
    0, () => RegisterPullTimeUpdateExclusionRequest$, () => RegisterPullTimeUpdateExclusionResponse$
];
exports.SetRepositoryPolicy$ = [9, n0, _SRP,
    0, () => SetRepositoryPolicyRequest$, () => SetRepositoryPolicyResponse$
];
exports.StartImageScan$ = [9, n0, _SIS,
    0, () => StartImageScanRequest$, () => StartImageScanResponse$
];
exports.StartLifecyclePolicyPreview$ = [9, n0, _SLPP,
    0, () => StartLifecyclePolicyPreviewRequest$, () => StartLifecyclePolicyPreviewResponse$
];
exports.TagResource$ = [9, n0, _TR,
    0, () => TagResourceRequest$, () => TagResourceResponse$
];
exports.UntagResource$ = [9, n0, _UR,
    0, () => UntagResourceRequest$, () => UntagResourceResponse$
];
exports.UpdateImageStorageClass$ = [9, n0, _UISC,
    0, () => UpdateImageStorageClassRequest$, () => UpdateImageStorageClassResponse$
];
exports.UpdatePullThroughCacheRule$ = [9, n0, _UPTCR,
    0, () => UpdatePullThroughCacheRuleRequest$, () => UpdatePullThroughCacheRuleResponse$
];
exports.UpdateRepositoryCreationTemplate$ = [9, n0, _URCT,
    0, () => UpdateRepositoryCreationTemplateRequest$, () => UpdateRepositoryCreationTemplateResponse$
];
exports.UploadLayerPart$ = [9, n0, _ULP,
    0, () => UploadLayerPartRequest$, () => UploadLayerPartResponse$
];
exports.ValidatePullThroughCacheRule$ = [9, n0, _VPTCR,
    0, () => ValidatePullThroughCacheRuleRequest$, () => ValidatePullThroughCacheRuleResponse$
];
