// @aws-sdk/client-apprunner/dist-cjs/index.js
var __exportStar = (m, e) => { Object.assign(e, m); };
const { getAwsRegionExtensionConfiguration, resolveAwsRegionExtensionConfiguration, resolveUserAgentConfig, resolveHostHeaderConfig, getUserAgentPlugin, getHostHeaderPlugin, getLoggerPlugin, getRecursionDetectionPlugin } = require("@aws-sdk/core/client");
const { getHttpAuthSchemeEndpointRuleSetPlugin, DefaultIdentityProviderConfig, getHttpSigningPlugin, createPaginator } = require("@smithy/core");
const { getDefaultExtensionConfiguration, resolveDefaultRuntimeConfig, Client, Command, createAggregatedClient } = require("@smithy/core/client");
exports.$Command = Command;
exports.__Client = Client;
const { resolveRegionConfig } = require("@smithy/core/config");
const { resolveEndpointConfig, getEndpointPlugin } = require("@smithy/core/endpoints");
const { getHttpHandlerExtensionConfiguration, resolveHttpHandlerRuntimeConfig, getContentLengthPlugin } = require("@smithy/core/protocols");
const { resolveRetryConfig, getRetryPlugin } = require("@smithy/core/retry");
const { getSchemaSerdePlugin } = require("@smithy/core/schema");
const { resolveHttpAuthSchemeConfig, defaultAppRunnerHttpAuthSchemeParametersProvider } = require("./auth/httpAuthSchemeProvider");
const { getRuntimeConfig } = require("./runtimeConfig");
const { AssociateCustomDomain$, CreateAutoScalingConfiguration$, CreateConnection$, CreateObservabilityConfiguration$, CreateService$, CreateVpcConnector$, CreateVpcIngressConnection$, DeleteAutoScalingConfiguration$, DeleteConnection$, DeleteObservabilityConfiguration$, DeleteService$, DeleteVpcConnector$, DeleteVpcIngressConnection$, DescribeAutoScalingConfiguration$, DescribeCustomDomains$, DescribeObservabilityConfiguration$, DescribeService$, DescribeVpcConnector$, DescribeVpcIngressConnection$, DisassociateCustomDomain$, ListAutoScalingConfigurations$, ListConnections$, ListObservabilityConfigurations$, ListOperations$, ListServices$, ListServicesForAutoScalingConfiguration$, ListTagsForResource$, ListVpcConnectors$, ListVpcIngressConnections$, PauseService$, ResumeService$, StartDeployment$, TagResource$, UntagResource$, UpdateDefaultAutoScalingConfiguration$, UpdateService$, UpdateVpcIngressConnection$ } = require("./schemas/schemas_0");
__exportStar(require("./schemas/schemas_0"), exports);
__exportStar(require("./models/errors"), exports);
const { AppRunnerServiceException } = require("./models/AppRunnerServiceException");
exports.AppRunnerServiceException = AppRunnerServiceException;

const resolveClientEndpointParameters = (options) => {
    return Object.assign(options, {
        useDualstackEndpoint: options.useDualstackEndpoint ?? false,
        useFipsEndpoint: options.useFipsEndpoint ?? false,
        defaultSigningName: "apprunner",
    });
};
const commonParams = {
    UseFIPS: { type: "builtInParams", name: "useFipsEndpoint" },
    Endpoint: { type: "builtInParams", name: "endpoint" },
    Region: { type: "builtInParams", name: "region" },
    UseDualStack: { type: "builtInParams", name: "useDualstackEndpoint" },
};

const getHttpAuthExtensionConfiguration = (runtimeConfig) => {
    const _httpAuthSchemes = runtimeConfig.httpAuthSchemes;
    let _httpAuthSchemeProvider = runtimeConfig.httpAuthSchemeProvider;
    let _credentials = runtimeConfig.credentials;
    return {
        setHttpAuthScheme(httpAuthScheme) {
            const index = _httpAuthSchemes.findIndex((scheme) => scheme.schemeId === httpAuthScheme.schemeId);
            if (index === -1) {
                _httpAuthSchemes.push(httpAuthScheme);
            }
            else {
                _httpAuthSchemes.splice(index, 1, httpAuthScheme);
            }
        },
        httpAuthSchemes() {
            return _httpAuthSchemes;
        },
        setHttpAuthSchemeProvider(httpAuthSchemeProvider) {
            _httpAuthSchemeProvider = httpAuthSchemeProvider;
        },
        httpAuthSchemeProvider() {
            return _httpAuthSchemeProvider;
        },
        setCredentials(credentials) {
            _credentials = credentials;
        },
        credentials() {
            return _credentials;
        },
    };
};
const resolveHttpAuthRuntimeConfig = (config) => {
    return {
        httpAuthSchemes: config.httpAuthSchemes(),
        httpAuthSchemeProvider: config.httpAuthSchemeProvider(),
        credentials: config.credentials(),
    };
};

const resolveRuntimeExtensions = (runtimeConfig, extensions) => {
    const extensionConfiguration = Object.assign(getAwsRegionExtensionConfiguration(runtimeConfig), getDefaultExtensionConfiguration(runtimeConfig), getHttpHandlerExtensionConfiguration(runtimeConfig), getHttpAuthExtensionConfiguration(runtimeConfig));
    extensions.forEach((extension) => extension.configure(extensionConfiguration));
    return Object.assign(runtimeConfig, resolveAwsRegionExtensionConfiguration(extensionConfiguration), resolveDefaultRuntimeConfig(extensionConfiguration), resolveHttpHandlerRuntimeConfig(extensionConfiguration), resolveHttpAuthRuntimeConfig(extensionConfiguration));
};

class AppRunnerClient extends Client {
    config;
    constructor(...[configuration]) {
        const _config_0 = getRuntimeConfig(configuration || {});
        super(_config_0);
        this.initConfig = _config_0;
        const _config_1 = resolveClientEndpointParameters(_config_0);
        const _config_2 = resolveUserAgentConfig(_config_1);
        const _config_3 = resolveRetryConfig(_config_2);
        const _config_4 = resolveRegionConfig(_config_3);
        const _config_5 = resolveHostHeaderConfig(_config_4);
        const _config_6 = resolveEndpointConfig(_config_5);
        const _config_7 = resolveHttpAuthSchemeConfig(_config_6);
        const _config_8 = resolveRuntimeExtensions(_config_7, configuration?.extensions || []);
        this.config = _config_8;
        this.middlewareStack.use(getSchemaSerdePlugin(this.config));
        this.middlewareStack.use(getUserAgentPlugin(this.config));
        this.middlewareStack.use(getRetryPlugin(this.config));
        this.middlewareStack.use(getContentLengthPlugin(this.config));
        this.middlewareStack.use(getHostHeaderPlugin(this.config));
        this.middlewareStack.use(getLoggerPlugin(this.config));
        this.middlewareStack.use(getRecursionDetectionPlugin(this.config));
        this.middlewareStack.use(getHttpAuthSchemeEndpointRuleSetPlugin(this.config, {
            httpAuthSchemeParametersProvider: defaultAppRunnerHttpAuthSchemeParametersProvider,
            identityProviderConfigProvider: async (config) => new DefaultIdentityProviderConfig({
                "aws.auth#sigv4": config.credentials,
            }),
        }));
        this.middlewareStack.use(getHttpSigningPlugin(this.config));
    }
    destroy() {
        super.destroy();
    }
}

class AssociateCustomDomainCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AppRunner", "AssociateCustomDomain", {})
    .n("AppRunnerClient", "AssociateCustomDomainCommand")
    .sc(AssociateCustomDomain$)
    .build() {
}

class CreateAutoScalingConfigurationCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AppRunner", "CreateAutoScalingConfiguration", {})
    .n("AppRunnerClient", "CreateAutoScalingConfigurationCommand")
    .sc(CreateAutoScalingConfiguration$)
    .build() {
}

class CreateConnectionCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AppRunner", "CreateConnection", {})
    .n("AppRunnerClient", "CreateConnectionCommand")
    .sc(CreateConnection$)
    .build() {
}

class CreateObservabilityConfigurationCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AppRunner", "CreateObservabilityConfiguration", {})
    .n("AppRunnerClient", "CreateObservabilityConfigurationCommand")
    .sc(CreateObservabilityConfiguration$)
    .build() {
}

class CreateServiceCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AppRunner", "CreateService", {})
    .n("AppRunnerClient", "CreateServiceCommand")
    .sc(CreateService$)
    .build() {
}

class CreateVpcConnectorCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AppRunner", "CreateVpcConnector", {})
    .n("AppRunnerClient", "CreateVpcConnectorCommand")
    .sc(CreateVpcConnector$)
    .build() {
}

class CreateVpcIngressConnectionCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AppRunner", "CreateVpcIngressConnection", {})
    .n("AppRunnerClient", "CreateVpcIngressConnectionCommand")
    .sc(CreateVpcIngressConnection$)
    .build() {
}

class DeleteAutoScalingConfigurationCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AppRunner", "DeleteAutoScalingConfiguration", {})
    .n("AppRunnerClient", "DeleteAutoScalingConfigurationCommand")
    .sc(DeleteAutoScalingConfiguration$)
    .build() {
}

class DeleteConnectionCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AppRunner", "DeleteConnection", {})
    .n("AppRunnerClient", "DeleteConnectionCommand")
    .sc(DeleteConnection$)
    .build() {
}

class DeleteObservabilityConfigurationCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AppRunner", "DeleteObservabilityConfiguration", {})
    .n("AppRunnerClient", "DeleteObservabilityConfigurationCommand")
    .sc(DeleteObservabilityConfiguration$)
    .build() {
}

class DeleteServiceCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AppRunner", "DeleteService", {})
    .n("AppRunnerClient", "DeleteServiceCommand")
    .sc(DeleteService$)
    .build() {
}

class DeleteVpcConnectorCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AppRunner", "DeleteVpcConnector", {})
    .n("AppRunnerClient", "DeleteVpcConnectorCommand")
    .sc(DeleteVpcConnector$)
    .build() {
}

class DeleteVpcIngressConnectionCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AppRunner", "DeleteVpcIngressConnection", {})
    .n("AppRunnerClient", "DeleteVpcIngressConnectionCommand")
    .sc(DeleteVpcIngressConnection$)
    .build() {
}

class DescribeAutoScalingConfigurationCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AppRunner", "DescribeAutoScalingConfiguration", {})
    .n("AppRunnerClient", "DescribeAutoScalingConfigurationCommand")
    .sc(DescribeAutoScalingConfiguration$)
    .build() {
}

class DescribeCustomDomainsCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AppRunner", "DescribeCustomDomains", {})
    .n("AppRunnerClient", "DescribeCustomDomainsCommand")
    .sc(DescribeCustomDomains$)
    .build() {
}

class DescribeObservabilityConfigurationCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AppRunner", "DescribeObservabilityConfiguration", {})
    .n("AppRunnerClient", "DescribeObservabilityConfigurationCommand")
    .sc(DescribeObservabilityConfiguration$)
    .build() {
}

class DescribeServiceCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AppRunner", "DescribeService", {})
    .n("AppRunnerClient", "DescribeServiceCommand")
    .sc(DescribeService$)
    .build() {
}

class DescribeVpcConnectorCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AppRunner", "DescribeVpcConnector", {})
    .n("AppRunnerClient", "DescribeVpcConnectorCommand")
    .sc(DescribeVpcConnector$)
    .build() {
}

class DescribeVpcIngressConnectionCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AppRunner", "DescribeVpcIngressConnection", {})
    .n("AppRunnerClient", "DescribeVpcIngressConnectionCommand")
    .sc(DescribeVpcIngressConnection$)
    .build() {
}

class DisassociateCustomDomainCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AppRunner", "DisassociateCustomDomain", {})
    .n("AppRunnerClient", "DisassociateCustomDomainCommand")
    .sc(DisassociateCustomDomain$)
    .build() {
}

class ListAutoScalingConfigurationsCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AppRunner", "ListAutoScalingConfigurations", {})
    .n("AppRunnerClient", "ListAutoScalingConfigurationsCommand")
    .sc(ListAutoScalingConfigurations$)
    .build() {
}

class ListConnectionsCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AppRunner", "ListConnections", {})
    .n("AppRunnerClient", "ListConnectionsCommand")
    .sc(ListConnections$)
    .build() {
}

class ListObservabilityConfigurationsCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AppRunner", "ListObservabilityConfigurations", {})
    .n("AppRunnerClient", "ListObservabilityConfigurationsCommand")
    .sc(ListObservabilityConfigurations$)
    .build() {
}

class ListOperationsCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AppRunner", "ListOperations", {})
    .n("AppRunnerClient", "ListOperationsCommand")
    .sc(ListOperations$)
    .build() {
}

class ListServicesCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AppRunner", "ListServices", {})
    .n("AppRunnerClient", "ListServicesCommand")
    .sc(ListServices$)
    .build() {
}

class ListServicesForAutoScalingConfigurationCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AppRunner", "ListServicesForAutoScalingConfiguration", {})
    .n("AppRunnerClient", "ListServicesForAutoScalingConfigurationCommand")
    .sc(ListServicesForAutoScalingConfiguration$)
    .build() {
}

class ListTagsForResourceCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AppRunner", "ListTagsForResource", {})
    .n("AppRunnerClient", "ListTagsForResourceCommand")
    .sc(ListTagsForResource$)
    .build() {
}

class ListVpcConnectorsCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AppRunner", "ListVpcConnectors", {})
    .n("AppRunnerClient", "ListVpcConnectorsCommand")
    .sc(ListVpcConnectors$)
    .build() {
}

class ListVpcIngressConnectionsCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AppRunner", "ListVpcIngressConnections", {})
    .n("AppRunnerClient", "ListVpcIngressConnectionsCommand")
    .sc(ListVpcIngressConnections$)
    .build() {
}

class PauseServiceCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AppRunner", "PauseService", {})
    .n("AppRunnerClient", "PauseServiceCommand")
    .sc(PauseService$)
    .build() {
}

class ResumeServiceCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AppRunner", "ResumeService", {})
    .n("AppRunnerClient", "ResumeServiceCommand")
    .sc(ResumeService$)
    .build() {
}

class StartDeploymentCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AppRunner", "StartDeployment", {})
    .n("AppRunnerClient", "StartDeploymentCommand")
    .sc(StartDeployment$)
    .build() {
}

class TagResourceCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AppRunner", "TagResource", {})
    .n("AppRunnerClient", "TagResourceCommand")
    .sc(TagResource$)
    .build() {
}

class UntagResourceCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AppRunner", "UntagResource", {})
    .n("AppRunnerClient", "UntagResourceCommand")
    .sc(UntagResource$)
    .build() {
}

class UpdateDefaultAutoScalingConfigurationCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AppRunner", "UpdateDefaultAutoScalingConfiguration", {})
    .n("AppRunnerClient", "UpdateDefaultAutoScalingConfigurationCommand")
    .sc(UpdateDefaultAutoScalingConfiguration$)
    .build() {
}

class UpdateServiceCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AppRunner", "UpdateService", {})
    .n("AppRunnerClient", "UpdateServiceCommand")
    .sc(UpdateService$)
    .build() {
}

class UpdateVpcIngressConnectionCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AppRunner", "UpdateVpcIngressConnection", {})
    .n("AppRunnerClient", "UpdateVpcIngressConnectionCommand")
    .sc(UpdateVpcIngressConnection$)
    .build() {
}

const paginateDescribeCustomDomains = createPaginator(AppRunnerClient, DescribeCustomDomainsCommand, "NextToken", "NextToken", "MaxResults");

const paginateListAutoScalingConfigurations = createPaginator(AppRunnerClient, ListAutoScalingConfigurationsCommand, "NextToken", "NextToken", "MaxResults");

const paginateListConnections = createPaginator(AppRunnerClient, ListConnectionsCommand, "NextToken", "NextToken", "MaxResults");

const paginateListObservabilityConfigurations = createPaginator(AppRunnerClient, ListObservabilityConfigurationsCommand, "NextToken", "NextToken", "MaxResults");

const paginateListOperations = createPaginator(AppRunnerClient, ListOperationsCommand, "NextToken", "NextToken", "MaxResults");

const paginateListServicesForAutoScalingConfiguration = createPaginator(AppRunnerClient, ListServicesForAutoScalingConfigurationCommand, "NextToken", "NextToken", "MaxResults");

const paginateListServices = createPaginator(AppRunnerClient, ListServicesCommand, "NextToken", "NextToken", "MaxResults");

const paginateListVpcConnectors = createPaginator(AppRunnerClient, ListVpcConnectorsCommand, "NextToken", "NextToken", "MaxResults");

const paginateListVpcIngressConnections = createPaginator(AppRunnerClient, ListVpcIngressConnectionsCommand, "NextToken", "NextToken", "MaxResults");

const commands = {
    AssociateCustomDomainCommand,
    CreateAutoScalingConfigurationCommand,
    CreateConnectionCommand,
    CreateObservabilityConfigurationCommand,
    CreateServiceCommand,
    CreateVpcConnectorCommand,
    CreateVpcIngressConnectionCommand,
    DeleteAutoScalingConfigurationCommand,
    DeleteConnectionCommand,
    DeleteObservabilityConfigurationCommand,
    DeleteServiceCommand,
    DeleteVpcConnectorCommand,
    DeleteVpcIngressConnectionCommand,
    DescribeAutoScalingConfigurationCommand,
    DescribeCustomDomainsCommand,
    DescribeObservabilityConfigurationCommand,
    DescribeServiceCommand,
    DescribeVpcConnectorCommand,
    DescribeVpcIngressConnectionCommand,
    DisassociateCustomDomainCommand,
    ListAutoScalingConfigurationsCommand,
    ListConnectionsCommand,
    ListObservabilityConfigurationsCommand,
    ListOperationsCommand,
    ListServicesCommand,
    ListServicesForAutoScalingConfigurationCommand,
    ListTagsForResourceCommand,
    ListVpcConnectorsCommand,
    ListVpcIngressConnectionsCommand,
    PauseServiceCommand,
    ResumeServiceCommand,
    StartDeploymentCommand,
    TagResourceCommand,
    UntagResourceCommand,
    UpdateDefaultAutoScalingConfigurationCommand,
    UpdateServiceCommand,
    UpdateVpcIngressConnectionCommand,
};
const paginators = {
    paginateDescribeCustomDomains,
    paginateListAutoScalingConfigurations,
    paginateListConnections,
    paginateListObservabilityConfigurations,
    paginateListOperations,
    paginateListServices,
    paginateListServicesForAutoScalingConfiguration,
    paginateListVpcConnectors,
    paginateListVpcIngressConnections,
};
class AppRunner extends AppRunnerClient {
}
createAggregatedClient(commands, AppRunner, { paginators });

const CertificateValidationRecordStatus = {
    FAILED: "FAILED",
    PENDING_VALIDATION: "PENDING_VALIDATION",
    SUCCESS: "SUCCESS",
};
const CustomDomainAssociationStatus = {
    ACTIVE: "ACTIVE",
    BINDING_CERTIFICATE: "BINDING_CERTIFICATE",
    CREATE_FAILED: "CREATE_FAILED",
    CREATING: "CREATING",
    DELETE_FAILED: "DELETE_FAILED",
    DELETING: "DELETING",
    PENDING_CERTIFICATE_DNS_VALIDATION: "PENDING_CERTIFICATE_DNS_VALIDATION",
};
const AutoScalingConfigurationStatus = {
    ACTIVE: "ACTIVE",
    INACTIVE: "INACTIVE",
};
const ProviderType = {
    BITBUCKET: "BITBUCKET",
    GITHUB: "GITHUB",
};
const ConnectionStatus = {
    AVAILABLE: "AVAILABLE",
    DELETED: "DELETED",
    ERROR: "ERROR",
    PENDING_HANDSHAKE: "PENDING_HANDSHAKE",
};
const TracingVendor = {
    AWSXRAY: "AWSXRAY",
};
const ObservabilityConfigurationStatus = {
    ACTIVE: "ACTIVE",
    INACTIVE: "INACTIVE",
};
const HealthCheckProtocol = {
    HTTP: "HTTP",
    TCP: "TCP",
};
const EgressType = {
    DEFAULT: "DEFAULT",
    VPC: "VPC",
};
const IpAddressType = {
    DUAL_STACK: "DUAL_STACK",
    IPV4: "IPV4",
};
const Runtime = {
    CORRETTO_11: "CORRETTO_11",
    CORRETTO_8: "CORRETTO_8",
    DOTNET_6: "DOTNET_6",
    GO_1: "GO_1",
    NODEJS_12: "NODEJS_12",
    NODEJS_14: "NODEJS_14",
    NODEJS_16: "NODEJS_16",
    NODEJS_18: "NODEJS_18",
    NODEJS_22: "NODEJS_22",
    PHP_81: "PHP_81",
    PYTHON_3: "PYTHON_3",
    PYTHON_311: "PYTHON_311",
    RUBY_31: "RUBY_31",
};
const ConfigurationSource = {
    API: "API",
    REPOSITORY: "REPOSITORY",
};
const SourceCodeVersionType = {
    BRANCH: "BRANCH",
};
const ImageRepositoryType = {
    ECR: "ECR",
    ECR_PUBLIC: "ECR_PUBLIC",
};
const ServiceStatus = {
    CREATE_FAILED: "CREATE_FAILED",
    DELETED: "DELETED",
    DELETE_FAILED: "DELETE_FAILED",
    OPERATION_IN_PROGRESS: "OPERATION_IN_PROGRESS",
    PAUSED: "PAUSED",
    RUNNING: "RUNNING",
};
const VpcConnectorStatus = {
    ACTIVE: "ACTIVE",
    INACTIVE: "INACTIVE",
};
const VpcIngressConnectionStatus = {
    AVAILABLE: "AVAILABLE",
    DELETED: "DELETED",
    FAILED_CREATION: "FAILED_CREATION",
    FAILED_DELETION: "FAILED_DELETION",
    FAILED_UPDATE: "FAILED_UPDATE",
    PENDING_CREATION: "PENDING_CREATION",
    PENDING_DELETION: "PENDING_DELETION",
    PENDING_UPDATE: "PENDING_UPDATE",
};
const OperationStatus = {
    FAILED: "FAILED",
    IN_PROGRESS: "IN_PROGRESS",
    PENDING: "PENDING",
    ROLLBACK_FAILED: "ROLLBACK_FAILED",
    ROLLBACK_IN_PROGRESS: "ROLLBACK_IN_PROGRESS",
    ROLLBACK_SUCCEEDED: "ROLLBACK_SUCCEEDED",
    SUCCEEDED: "SUCCEEDED",
};
const OperationType = {
    CREATE_SERVICE: "CREATE_SERVICE",
    DELETE_SERVICE: "DELETE_SERVICE",
    PAUSE_SERVICE: "PAUSE_SERVICE",
    RESUME_SERVICE: "RESUME_SERVICE",
    START_DEPLOYMENT: "START_DEPLOYMENT",
    UPDATE_SERVICE: "UPDATE_SERVICE",
};

exports.AppRunner = AppRunner;
exports.AppRunnerClient = AppRunnerClient;
exports.AssociateCustomDomainCommand = AssociateCustomDomainCommand;
exports.AutoScalingConfigurationStatus = AutoScalingConfigurationStatus;
exports.CertificateValidationRecordStatus = CertificateValidationRecordStatus;
exports.ConfigurationSource = ConfigurationSource;
exports.ConnectionStatus = ConnectionStatus;
exports.CreateAutoScalingConfigurationCommand = CreateAutoScalingConfigurationCommand;
exports.CreateConnectionCommand = CreateConnectionCommand;
exports.CreateObservabilityConfigurationCommand = CreateObservabilityConfigurationCommand;
exports.CreateServiceCommand = CreateServiceCommand;
exports.CreateVpcConnectorCommand = CreateVpcConnectorCommand;
exports.CreateVpcIngressConnectionCommand = CreateVpcIngressConnectionCommand;
exports.CustomDomainAssociationStatus = CustomDomainAssociationStatus;
exports.DeleteAutoScalingConfigurationCommand = DeleteAutoScalingConfigurationCommand;
exports.DeleteConnectionCommand = DeleteConnectionCommand;
exports.DeleteObservabilityConfigurationCommand = DeleteObservabilityConfigurationCommand;
exports.DeleteServiceCommand = DeleteServiceCommand;
exports.DeleteVpcConnectorCommand = DeleteVpcConnectorCommand;
exports.DeleteVpcIngressConnectionCommand = DeleteVpcIngressConnectionCommand;
exports.DescribeAutoScalingConfigurationCommand = DescribeAutoScalingConfigurationCommand;
exports.DescribeCustomDomainsCommand = DescribeCustomDomainsCommand;
exports.DescribeObservabilityConfigurationCommand = DescribeObservabilityConfigurationCommand;
exports.DescribeServiceCommand = DescribeServiceCommand;
exports.DescribeVpcConnectorCommand = DescribeVpcConnectorCommand;
exports.DescribeVpcIngressConnectionCommand = DescribeVpcIngressConnectionCommand;
exports.DisassociateCustomDomainCommand = DisassociateCustomDomainCommand;
exports.EgressType = EgressType;
exports.HealthCheckProtocol = HealthCheckProtocol;
exports.ImageRepositoryType = ImageRepositoryType;
exports.IpAddressType = IpAddressType;
exports.ListAutoScalingConfigurationsCommand = ListAutoScalingConfigurationsCommand;
exports.ListConnectionsCommand = ListConnectionsCommand;
exports.ListObservabilityConfigurationsCommand = ListObservabilityConfigurationsCommand;
exports.ListOperationsCommand = ListOperationsCommand;
exports.ListServicesCommand = ListServicesCommand;
exports.ListServicesForAutoScalingConfigurationCommand = ListServicesForAutoScalingConfigurationCommand;
exports.ListTagsForResourceCommand = ListTagsForResourceCommand;
exports.ListVpcConnectorsCommand = ListVpcConnectorsCommand;
exports.ListVpcIngressConnectionsCommand = ListVpcIngressConnectionsCommand;
exports.ObservabilityConfigurationStatus = ObservabilityConfigurationStatus;
exports.OperationStatus = OperationStatus;
exports.OperationType = OperationType;
exports.PauseServiceCommand = PauseServiceCommand;
exports.ProviderType = ProviderType;
exports.ResumeServiceCommand = ResumeServiceCommand;
exports.Runtime = Runtime;
exports.ServiceStatus = ServiceStatus;
exports.SourceCodeVersionType = SourceCodeVersionType;
exports.StartDeploymentCommand = StartDeploymentCommand;
exports.TagResourceCommand = TagResourceCommand;
exports.TracingVendor = TracingVendor;
exports.UntagResourceCommand = UntagResourceCommand;
exports.UpdateDefaultAutoScalingConfigurationCommand = UpdateDefaultAutoScalingConfigurationCommand;
exports.UpdateServiceCommand = UpdateServiceCommand;
exports.UpdateVpcIngressConnectionCommand = UpdateVpcIngressConnectionCommand;
exports.VpcConnectorStatus = VpcConnectorStatus;
exports.VpcIngressConnectionStatus = VpcIngressConnectionStatus;
exports.paginateDescribeCustomDomains = paginateDescribeCustomDomains;
exports.paginateListAutoScalingConfigurations = paginateListAutoScalingConfigurations;
exports.paginateListConnections = paginateListConnections;
exports.paginateListObservabilityConfigurations = paginateListObservabilityConfigurations;
exports.paginateListOperations = paginateListOperations;
exports.paginateListServices = paginateListServices;
exports.paginateListServicesForAutoScalingConfiguration = paginateListServicesForAutoScalingConfiguration;
exports.paginateListVpcConnectors = paginateListVpcConnectors;
exports.paginateListVpcIngressConnections = paginateListVpcIngressConnections;
